// @lit-protocol/lit-third-party-libs/src/lib/lit-third-party-libs.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LitThirdPartyLibs = void 0;
// @ts-nocheck
exports.LitThirdPartyLibs = (() => {
    var __create = Object.create;
    var __defProp = Object.defineProperty;
    var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
    var __getOwnPropNames = Object.getOwnPropertyNames;
    var __getProtoOf = Object.getPrototypeOf;
    var __hasOwnProp = Object.prototype.hasOwnProperty;
    var __commonJS = (cb, mod2) => function __require() {
        return (mod2 ||
            (0, cb[__getOwnPropNames(cb)[0]])((mod2 = { exports: {} }).exports, mod2),
            mod2.exports);
    };
    var __export = (target, all2) => {
        for (var name4 in all2)
            __defProp(target, name4, { get: all2[name4], enumerable: true });
    };
    var __copyProps = (to, from3, except, desc) => {
        if ((from3 && typeof from3 === 'object') || typeof from3 === 'function') {
            for (let key of __getOwnPropNames(from3))
                if (!__hasOwnProp.call(to, key) && key !== except)
                    __defProp(to, key, {
                        get: () => from3[key],
                        enumerable: !(desc = __getOwnPropDesc(from3, key)) || desc.enumerable,
                    });
        }
        return to;
    };
    var __toESM = (mod2, isNodeMode, target) => ((target = mod2 != null ? __create(__getProtoOf(mod2)) : {}),
        __copyProps(
        // If the importer is in node compatibility mode or this is not an ESM
        // file that has been converted to a CommonJS file using a Babel-
        // compatible transform (i.e. "__esModule" has not been set), then set
        // "default" to the CommonJS "module.exports" for node compatibility.
        isNodeMode || !mod2 || !mod2.__esModule
            ? __defProp(target, 'default', { value: mod2, enumerable: true })
            : target, mod2));
    var __toCommonJS = (mod2) => __copyProps(__defProp({}, '__esModule', { value: true }), mod2);
    // node_modules/is-plain-obj/index.js
    var require_is_plain_obj = __commonJS({
        'node_modules/is-plain-obj/index.js'(exports2, module2) {
            'use strict';
            module2.exports = (value) => {
                if (Object.prototype.toString.call(value) !== '[object Object]') {
                    return false;
                }
                const prototype = Object.getPrototypeOf(value);
                return prototype === null || prototype === Object.prototype;
            };
        },
    });
    // node_modules/merge-options/index.js
    var require_merge_options = __commonJS({
        'node_modules/merge-options/index.js'(exports2, module2) {
            'use strict';
            var isOptionObject = require_is_plain_obj();
            var { hasOwnProperty } = Object.prototype;
            var { propertyIsEnumerable } = Object;
            var defineProperty = (object, name4, value) => Object.defineProperty(object, name4, {
                value,
                writable: true,
                enumerable: true,
                configurable: true,
            });
            var globalThis2 = exports2;
            var defaultMergeOptions = {
                concatArrays: false,
                ignoreUndefined: false,
            };
            var getEnumerableOwnPropertyKeys = (value) => {
                const keys = [];
                for (const key in value) {
                    if (hasOwnProperty.call(value, key)) {
                        keys.push(key);
                    }
                }
                if (Object.getOwnPropertySymbols) {
                    const symbols = Object.getOwnPropertySymbols(value);
                    for (const symbol2 of symbols) {
                        if (propertyIsEnumerable.call(value, symbol2)) {
                            keys.push(symbol2);
                        }
                    }
                }
                return keys;
            };
            function clone(value) {
                if (Array.isArray(value)) {
                    return cloneArray(value);
                }
                if (isOptionObject(value)) {
                    return cloneOptionObject(value);
                }
                return value;
            }
            function cloneArray(array) {
                const result = array.slice(0, 0);
                getEnumerableOwnPropertyKeys(array).forEach((key) => {
                    defineProperty(result, key, clone(array[key]));
                });
                return result;
            }
            function cloneOptionObject(object) {
                const result = Object.getPrototypeOf(object) === null
                    ? /* @__PURE__ */ Object.create(null)
                    : {};
                getEnumerableOwnPropertyKeys(object).forEach((key) => {
                    defineProperty(result, key, clone(object[key]));
                });
                return result;
            }
            var mergeKeys = (merged, source, keys, config) => {
                keys.forEach((key) => {
                    if (typeof source[key] === 'undefined' && config.ignoreUndefined) {
                        return;
                    }
                    if (key in merged && merged[key] !== Object.getPrototypeOf(merged)) {
                        defineProperty(merged, key, merge(merged[key], source[key], config));
                    }
                    else {
                        defineProperty(merged, key, clone(source[key]));
                    }
                });
                return merged;
            };
            var concatArrays = (merged, source, config) => {
                let result = merged.slice(0, 0);
                let resultIndex = 0;
                [merged, source].forEach((array) => {
                    const indices = [];
                    for (let k = 0; k < array.length; k++) {
                        if (!hasOwnProperty.call(array, k)) {
                            continue;
                        }
                        indices.push(String(k));
                        if (array === merged) {
                            defineProperty(result, resultIndex++, array[k]);
                        }
                        else {
                            defineProperty(result, resultIndex++, clone(array[k]));
                        }
                    }
                    result = mergeKeys(result, array, getEnumerableOwnPropertyKeys(array).filter((key) => !indices.includes(key)), config);
                });
                return result;
            };
            function merge(merged, source, config) {
                if (config.concatArrays &&
                    Array.isArray(merged) &&
                    Array.isArray(source)) {
                    return concatArrays(merged, source, config);
                }
                if (!isOptionObject(source) || !isOptionObject(merged)) {
                    return clone(source);
                }
                return mergeKeys(merged, source, getEnumerableOwnPropertyKeys(source), config);
            }
            module2.exports = function (...options) {
                const config = merge(clone(defaultMergeOptions), (this !== globalThis2 && this) || {}, defaultMergeOptions);
                let merged = { _: {} };
                for (const option of options) {
                    if (option === void 0) {
                        continue;
                    }
                    if (!isOptionObject(option)) {
                        throw new TypeError('`' + option + '` is not an Option Object');
                    }
                    merged = merge(merged, { _: option }, config);
                }
                return merged._;
            };
        },
    });
    // node_modules/murmurhash3js-revisited/lib/murmurHash3js.js
    var require_murmurHash3js = __commonJS({
        'node_modules/murmurhash3js-revisited/lib/murmurHash3js.js'(exports2, module2) {
            (function (root, undefined2) {
                'use strict';
                var library = {
                    version: '3.0.0',
                    x86: {},
                    x64: {},
                    inputValidation: true,
                };
                function _validBytes(bytes) {
                    if (!Array.isArray(bytes) && !ArrayBuffer.isView(bytes)) {
                        return false;
                    }
                    for (var i = 0; i < bytes.length; i++) {
                        if (!Number.isInteger(bytes[i]) || bytes[i] < 0 || bytes[i] > 255) {
                            return false;
                        }
                    }
                    return true;
                }
                function _x86Multiply(m, n) {
                    return (m & 65535) * n + ((((m >>> 16) * n) & 65535) << 16);
                }
                function _x86Rotl(m, n) {
                    return (m << n) | (m >>> (32 - n));
                }
                function _x86Fmix(h) {
                    h ^= h >>> 16;
                    h = _x86Multiply(h, 2246822507);
                    h ^= h >>> 13;
                    h = _x86Multiply(h, 3266489909);
                    h ^= h >>> 16;
                    return h;
                }
                function _x64Add(m, n) {
                    m = [m[0] >>> 16, m[0] & 65535, m[1] >>> 16, m[1] & 65535];
                    n = [n[0] >>> 16, n[0] & 65535, n[1] >>> 16, n[1] & 65535];
                    var o = [0, 0, 0, 0];
                    o[3] += m[3] + n[3];
                    o[2] += o[3] >>> 16;
                    o[3] &= 65535;
                    o[2] += m[2] + n[2];
                    o[1] += o[2] >>> 16;
                    o[2] &= 65535;
                    o[1] += m[1] + n[1];
                    o[0] += o[1] >>> 16;
                    o[1] &= 65535;
                    o[0] += m[0] + n[0];
                    o[0] &= 65535;
                    return [(o[0] << 16) | o[1], (o[2] << 16) | o[3]];
                }
                function _x64Multiply(m, n) {
                    m = [m[0] >>> 16, m[0] & 65535, m[1] >>> 16, m[1] & 65535];
                    n = [n[0] >>> 16, n[0] & 65535, n[1] >>> 16, n[1] & 65535];
                    var o = [0, 0, 0, 0];
                    o[3] += m[3] * n[3];
                    o[2] += o[3] >>> 16;
                    o[3] &= 65535;
                    o[2] += m[2] * n[3];
                    o[1] += o[2] >>> 16;
                    o[2] &= 65535;
                    o[2] += m[3] * n[2];
                    o[1] += o[2] >>> 16;
                    o[2] &= 65535;
                    o[1] += m[1] * n[3];
                    o[0] += o[1] >>> 16;
                    o[1] &= 65535;
                    o[1] += m[2] * n[2];
                    o[0] += o[1] >>> 16;
                    o[1] &= 65535;
                    o[1] += m[3] * n[1];
                    o[0] += o[1] >>> 16;
                    o[1] &= 65535;
                    o[0] += m[0] * n[3] + m[1] * n[2] + m[2] * n[1] + m[3] * n[0];
                    o[0] &= 65535;
                    return [(o[0] << 16) | o[1], (o[2] << 16) | o[3]];
                }
                function _x64Rotl(m, n) {
                    n %= 64;
                    if (n === 32) {
                        return [m[1], m[0]];
                    }
                    else if (n < 32) {
                        return [
                            (m[0] << n) | (m[1] >>> (32 - n)),
                            (m[1] << n) | (m[0] >>> (32 - n)),
                        ];
                    }
                    else {
                        n -= 32;
                        return [
                            (m[1] << n) | (m[0] >>> (32 - n)),
                            (m[0] << n) | (m[1] >>> (32 - n)),
                        ];
                    }
                }
                function _x64LeftShift(m, n) {
                    n %= 64;
                    if (n === 0) {
                        return m;
                    }
                    else if (n < 32) {
                        return [(m[0] << n) | (m[1] >>> (32 - n)), m[1] << n];
                    }
                    else {
                        return [m[1] << (n - 32), 0];
                    }
                }
                function _x64Xor(m, n) {
                    return [m[0] ^ n[0], m[1] ^ n[1]];
                }
                function _x64Fmix(h) {
                    h = _x64Xor(h, [0, h[0] >>> 1]);
                    h = _x64Multiply(h, [4283543511, 3981806797]);
                    h = _x64Xor(h, [0, h[0] >>> 1]);
                    h = _x64Multiply(h, [3301882366, 444984403]);
                    h = _x64Xor(h, [0, h[0] >>> 1]);
                    return h;
                }
                library.x86.hash32 = function (bytes, seed) {
                    if (library.inputValidation && !_validBytes(bytes)) {
                        return undefined2;
                    }
                    seed = seed || 0;
                    var remainder = bytes.length % 4;
                    var blocks = bytes.length - remainder;
                    var h1 = seed;
                    var k1 = 0;
                    var c1 = 3432918353;
                    var c2 = 461845907;
                    for (var i = 0; i < blocks; i = i + 4) {
                        k1 =
                            bytes[i] |
                                (bytes[i + 1] << 8) |
                                (bytes[i + 2] << 16) |
                                (bytes[i + 3] << 24);
                        k1 = _x86Multiply(k1, c1);
                        k1 = _x86Rotl(k1, 15);
                        k1 = _x86Multiply(k1, c2);
                        h1 ^= k1;
                        h1 = _x86Rotl(h1, 13);
                        h1 = _x86Multiply(h1, 5) + 3864292196;
                    }
                    k1 = 0;
                    switch (remainder) {
                        case 3:
                            k1 ^= bytes[i + 2] << 16;
                        case 2:
                            k1 ^= bytes[i + 1] << 8;
                        case 1:
                            k1 ^= bytes[i];
                            k1 = _x86Multiply(k1, c1);
                            k1 = _x86Rotl(k1, 15);
                            k1 = _x86Multiply(k1, c2);
                            h1 ^= k1;
                    }
                    h1 ^= bytes.length;
                    h1 = _x86Fmix(h1);
                    return h1 >>> 0;
                };
                library.x86.hash128 = function (bytes, seed) {
                    if (library.inputValidation && !_validBytes(bytes)) {
                        return undefined2;
                    }
                    seed = seed || 0;
                    var remainder = bytes.length % 16;
                    var blocks = bytes.length - remainder;
                    var h1 = seed;
                    var h2 = seed;
                    var h3 = seed;
                    var h4 = seed;
                    var k1 = 0;
                    var k2 = 0;
                    var k3 = 0;
                    var k4 = 0;
                    var c1 = 597399067;
                    var c2 = 2869860233;
                    var c3 = 951274213;
                    var c4 = 2716044179;
                    for (var i = 0; i < blocks; i = i + 16) {
                        k1 =
                            bytes[i] |
                                (bytes[i + 1] << 8) |
                                (bytes[i + 2] << 16) |
                                (bytes[i + 3] << 24);
                        k2 =
                            bytes[i + 4] |
                                (bytes[i + 5] << 8) |
                                (bytes[i + 6] << 16) |
                                (bytes[i + 7] << 24);
                        k3 =
                            bytes[i + 8] |
                                (bytes[i + 9] << 8) |
                                (bytes[i + 10] << 16) |
                                (bytes[i + 11] << 24);
                        k4 =
                            bytes[i + 12] |
                                (bytes[i + 13] << 8) |
                                (bytes[i + 14] << 16) |
                                (bytes[i + 15] << 24);
                        k1 = _x86Multiply(k1, c1);
                        k1 = _x86Rotl(k1, 15);
                        k1 = _x86Multiply(k1, c2);
                        h1 ^= k1;
                        h1 = _x86Rotl(h1, 19);
                        h1 += h2;
                        h1 = _x86Multiply(h1, 5) + 1444728091;
                        k2 = _x86Multiply(k2, c2);
                        k2 = _x86Rotl(k2, 16);
                        k2 = _x86Multiply(k2, c3);
                        h2 ^= k2;
                        h2 = _x86Rotl(h2, 17);
                        h2 += h3;
                        h2 = _x86Multiply(h2, 5) + 197830471;
                        k3 = _x86Multiply(k3, c3);
                        k3 = _x86Rotl(k3, 17);
                        k3 = _x86Multiply(k3, c4);
                        h3 ^= k3;
                        h3 = _x86Rotl(h3, 15);
                        h3 += h4;
                        h3 = _x86Multiply(h3, 5) + 2530024501;
                        k4 = _x86Multiply(k4, c4);
                        k4 = _x86Rotl(k4, 18);
                        k4 = _x86Multiply(k4, c1);
                        h4 ^= k4;
                        h4 = _x86Rotl(h4, 13);
                        h4 += h1;
                        h4 = _x86Multiply(h4, 5) + 850148119;
                    }
                    k1 = 0;
                    k2 = 0;
                    k3 = 0;
                    k4 = 0;
                    switch (remainder) {
                        case 15:
                            k4 ^= bytes[i + 14] << 16;
                        case 14:
                            k4 ^= bytes[i + 13] << 8;
                        case 13:
                            k4 ^= bytes[i + 12];
                            k4 = _x86Multiply(k4, c4);
                            k4 = _x86Rotl(k4, 18);
                            k4 = _x86Multiply(k4, c1);
                            h4 ^= k4;
                        case 12:
                            k3 ^= bytes[i + 11] << 24;
                        case 11:
                            k3 ^= bytes[i + 10] << 16;
                        case 10:
                            k3 ^= bytes[i + 9] << 8;
                        case 9:
                            k3 ^= bytes[i + 8];
                            k3 = _x86Multiply(k3, c3);
                            k3 = _x86Rotl(k3, 17);
                            k3 = _x86Multiply(k3, c4);
                            h3 ^= k3;
                        case 8:
                            k2 ^= bytes[i + 7] << 24;
                        case 7:
                            k2 ^= bytes[i + 6] << 16;
                        case 6:
                            k2 ^= bytes[i + 5] << 8;
                        case 5:
                            k2 ^= bytes[i + 4];
                            k2 = _x86Multiply(k2, c2);
                            k2 = _x86Rotl(k2, 16);
                            k2 = _x86Multiply(k2, c3);
                            h2 ^= k2;
                        case 4:
                            k1 ^= bytes[i + 3] << 24;
                        case 3:
                            k1 ^= bytes[i + 2] << 16;
                        case 2:
                            k1 ^= bytes[i + 1] << 8;
                        case 1:
                            k1 ^= bytes[i];
                            k1 = _x86Multiply(k1, c1);
                            k1 = _x86Rotl(k1, 15);
                            k1 = _x86Multiply(k1, c2);
                            h1 ^= k1;
                    }
                    h1 ^= bytes.length;
                    h2 ^= bytes.length;
                    h3 ^= bytes.length;
                    h4 ^= bytes.length;
                    h1 += h2;
                    h1 += h3;
                    h1 += h4;
                    h2 += h1;
                    h3 += h1;
                    h4 += h1;
                    h1 = _x86Fmix(h1);
                    h2 = _x86Fmix(h2);
                    h3 = _x86Fmix(h3);
                    h4 = _x86Fmix(h4);
                    h1 += h2;
                    h1 += h3;
                    h1 += h4;
                    h2 += h1;
                    h3 += h1;
                    h4 += h1;
                    return (('00000000' + (h1 >>> 0).toString(16)).slice(-8) +
                        ('00000000' + (h2 >>> 0).toString(16)).slice(-8) +
                        ('00000000' + (h3 >>> 0).toString(16)).slice(-8) +
                        ('00000000' + (h4 >>> 0).toString(16)).slice(-8));
                };
                library.x64.hash128 = function (bytes, seed) {
                    if (library.inputValidation && !_validBytes(bytes)) {
                        return undefined2;
                    }
                    seed = seed || 0;
                    var remainder = bytes.length % 16;
                    var blocks = bytes.length - remainder;
                    var h1 = [0, seed];
                    var h2 = [0, seed];
                    var k1 = [0, 0];
                    var k2 = [0, 0];
                    var c1 = [2277735313, 289559509];
                    var c2 = [1291169091, 658871167];
                    for (var i = 0; i < blocks; i = i + 16) {
                        k1 = [
                            bytes[i + 4] |
                                (bytes[i + 5] << 8) |
                                (bytes[i + 6] << 16) |
                                (bytes[i + 7] << 24),
                            bytes[i] |
                                (bytes[i + 1] << 8) |
                                (bytes[i + 2] << 16) |
                                (bytes[i + 3] << 24),
                        ];
                        k2 = [
                            bytes[i + 12] |
                                (bytes[i + 13] << 8) |
                                (bytes[i + 14] << 16) |
                                (bytes[i + 15] << 24),
                            bytes[i + 8] |
                                (bytes[i + 9] << 8) |
                                (bytes[i + 10] << 16) |
                                (bytes[i + 11] << 24),
                        ];
                        k1 = _x64Multiply(k1, c1);
                        k1 = _x64Rotl(k1, 31);
                        k1 = _x64Multiply(k1, c2);
                        h1 = _x64Xor(h1, k1);
                        h1 = _x64Rotl(h1, 27);
                        h1 = _x64Add(h1, h2);
                        h1 = _x64Add(_x64Multiply(h1, [0, 5]), [0, 1390208809]);
                        k2 = _x64Multiply(k2, c2);
                        k2 = _x64Rotl(k2, 33);
                        k2 = _x64Multiply(k2, c1);
                        h2 = _x64Xor(h2, k2);
                        h2 = _x64Rotl(h2, 31);
                        h2 = _x64Add(h2, h1);
                        h2 = _x64Add(_x64Multiply(h2, [0, 5]), [0, 944331445]);
                    }
                    k1 = [0, 0];
                    k2 = [0, 0];
                    switch (remainder) {
                        case 15:
                            k2 = _x64Xor(k2, _x64LeftShift([0, bytes[i + 14]], 48));
                        case 14:
                            k2 = _x64Xor(k2, _x64LeftShift([0, bytes[i + 13]], 40));
                        case 13:
                            k2 = _x64Xor(k2, _x64LeftShift([0, bytes[i + 12]], 32));
                        case 12:
                            k2 = _x64Xor(k2, _x64LeftShift([0, bytes[i + 11]], 24));
                        case 11:
                            k2 = _x64Xor(k2, _x64LeftShift([0, bytes[i + 10]], 16));
                        case 10:
                            k2 = _x64Xor(k2, _x64LeftShift([0, bytes[i + 9]], 8));
                        case 9:
                            k2 = _x64Xor(k2, [0, bytes[i + 8]]);
                            k2 = _x64Multiply(k2, c2);
                            k2 = _x64Rotl(k2, 33);
                            k2 = _x64Multiply(k2, c1);
                            h2 = _x64Xor(h2, k2);
                        case 8:
                            k1 = _x64Xor(k1, _x64LeftShift([0, bytes[i + 7]], 56));
                        case 7:
                            k1 = _x64Xor(k1, _x64LeftShift([0, bytes[i + 6]], 48));
                        case 6:
                            k1 = _x64Xor(k1, _x64LeftShift([0, bytes[i + 5]], 40));
                        case 5:
                            k1 = _x64Xor(k1, _x64LeftShift([0, bytes[i + 4]], 32));
                        case 4:
                            k1 = _x64Xor(k1, _x64LeftShift([0, bytes[i + 3]], 24));
                        case 3:
                            k1 = _x64Xor(k1, _x64LeftShift([0, bytes[i + 2]], 16));
                        case 2:
                            k1 = _x64Xor(k1, _x64LeftShift([0, bytes[i + 1]], 8));
                        case 1:
                            k1 = _x64Xor(k1, [0, bytes[i]]);
                            k1 = _x64Multiply(k1, c1);
                            k1 = _x64Rotl(k1, 31);
                            k1 = _x64Multiply(k1, c2);
                            h1 = _x64Xor(h1, k1);
                    }
                    h1 = _x64Xor(h1, [0, bytes.length]);
                    h2 = _x64Xor(h2, [0, bytes.length]);
                    h1 = _x64Add(h1, h2);
                    h2 = _x64Add(h2, h1);
                    h1 = _x64Fmix(h1);
                    h2 = _x64Fmix(h2);
                    h1 = _x64Add(h1, h2);
                    h2 = _x64Add(h2, h1);
                    return (('00000000' + (h1[0] >>> 0).toString(16)).slice(-8) +
                        ('00000000' + (h1[1] >>> 0).toString(16)).slice(-8) +
                        ('00000000' + (h2[0] >>> 0).toString(16)).slice(-8) +
                        ('00000000' + (h2[1] >>> 0).toString(16)).slice(-8));
                };
                if (typeof exports2 !== 'undefined') {
                    if (typeof module2 !== 'undefined' && module2.exports) {
                        exports2 = module2.exports = library;
                    }
                    exports2.murmurHash3 = library;
                }
                else if (typeof define === 'function' && define.amd) {
                    define([], function () {
                        return library;
                    });
                }
                else {
                    library._murmurHash3 = root.murmurHash3;
                    library.noConflict = function () {
                        root.murmurHash3 = library._murmurHash3;
                        library._murmurHash3 = undefined2;
                        library.noConflict = undefined2;
                        return library;
                    };
                    root.murmurHash3 = library;
                }
            })(exports2);
        },
    });
    // node_modules/murmurhash3js-revisited/index.js
    var require_murmurhash3js_revisited = __commonJS({
        'node_modules/murmurhash3js-revisited/index.js'(exports2, module2) {
            module2.exports = require_murmurHash3js();
        },
    });
    // node_modules/err-code/index.js
    var require_err_code = __commonJS({
        'node_modules/err-code/index.js'(exports2, module2) {
            'use strict';
            function assign(obj, props) {
                for (const key in props) {
                    Object.defineProperty(obj, key, {
                        value: props[key],
                        enumerable: true,
                        configurable: true,
                    });
                }
                return obj;
            }
            function createError(err, code4, props) {
                if (!err || typeof err === 'string') {
                    throw new TypeError('Please pass an Error to err-code');
                }
                if (!props) {
                    props = {};
                }
                if (typeof code4 === 'object') {
                    props = code4;
                    code4 = '';
                }
                if (code4) {
                    props.code = code4;
                }
                try {
                    return assign(err, props);
                }
                catch (_) {
                    props.message = err.message;
                    props.stack = err.stack;
                    const ErrClass = function () { };
                    ErrClass.prototype = Object.create(Object.getPrototypeOf(err));
                    const output = assign(new ErrClass(), props);
                    return output;
                }
            }
            module2.exports = createError;
        },
    });
    // node_modules/@protobufjs/aspromise/index.js
    var require_aspromise = __commonJS({
        'node_modules/@protobufjs/aspromise/index.js'(exports2, module2) {
            'use strict';
            module2.exports = asPromise;
            function asPromise(fn, ctx) {
                var params = new Array(arguments.length - 1), offset = 0, index = 2, pending = true;
                while (index < arguments.length)
                    params[offset++] = arguments[index++];
                return new Promise(function executor(resolve, reject) {
                    params[offset] = function callback(err) {
                        if (pending) {
                            pending = false;
                            if (err)
                                reject(err);
                            else {
                                var params2 = new Array(arguments.length - 1), offset2 = 0;
                                while (offset2 < params2.length)
                                    params2[offset2++] = arguments[offset2];
                                resolve.apply(null, params2);
                            }
                        }
                    };
                    try {
                        fn.apply(ctx || null, params);
                    }
                    catch (err) {
                        if (pending) {
                            pending = false;
                            reject(err);
                        }
                    }
                });
            }
        },
    });
    // node_modules/@protobufjs/base64/index.js
    var require_base64 = __commonJS({
        'node_modules/@protobufjs/base64/index.js'(exports2) {
            'use strict';
            var base642 = exports2;
            base642.length = function length2(string2) {
                var p = string2.length;
                if (!p)
                    return 0;
                var n = 0;
                while (--p % 4 > 1 && string2.charAt(p) === '=')
                    ++n;
                return Math.ceil(string2.length * 3) / 4 - n;
            };
            var b64 = new Array(64);
            var s64 = new Array(123);
            for (i = 0; i < 64;)
                s64[(b64[i] =
                    i < 26 ? i + 65 : i < 52 ? i + 71 : i < 62 ? i - 4 : (i - 59) | 43)] = i++;
            var i;
            base642.encode = function encode7(buffer, start, end) {
                var parts = null, chunk = [];
                var i2 = 0, j = 0, t;
                while (start < end) {
                    var b = buffer[start++];
                    switch (j) {
                        case 0:
                            chunk[i2++] = b64[b >> 2];
                            t = (b & 3) << 4;
                            j = 1;
                            break;
                        case 1:
                            chunk[i2++] = b64[t | (b >> 4)];
                            t = (b & 15) << 2;
                            j = 2;
                            break;
                        case 2:
                            chunk[i2++] = b64[t | (b >> 6)];
                            chunk[i2++] = b64[b & 63];
                            j = 0;
                            break;
                    }
                    if (i2 > 8191) {
                        (parts || (parts = [])).push(String.fromCharCode.apply(String, chunk));
                        i2 = 0;
                    }
                }
                if (j) {
                    chunk[i2++] = b64[t];
                    chunk[i2++] = 61;
                    if (j === 1)
                        chunk[i2++] = 61;
                }
                if (parts) {
                    if (i2)
                        parts.push(String.fromCharCode.apply(String, chunk.slice(0, i2)));
                    return parts.join('');
                }
                return String.fromCharCode.apply(String, chunk.slice(0, i2));
            };
            var invalidEncoding = 'invalid encoding';
            base642.decode = function decode8(string2, buffer, offset) {
                var start = offset;
                var j = 0, t;
                for (var i2 = 0; i2 < string2.length;) {
                    var c = string2.charCodeAt(i2++);
                    if (c === 61 && j > 1)
                        break;
                    if ((c = s64[c]) === void 0)
                        throw Error(invalidEncoding);
                    switch (j) {
                        case 0:
                            t = c;
                            j = 1;
                            break;
                        case 1:
                            buffer[offset++] = (t << 2) | ((c & 48) >> 4);
                            t = c;
                            j = 2;
                            break;
                        case 2:
                            buffer[offset++] = ((t & 15) << 4) | ((c & 60) >> 2);
                            t = c;
                            j = 3;
                            break;
                        case 3:
                            buffer[offset++] = ((t & 3) << 6) | c;
                            j = 0;
                            break;
                    }
                }
                if (j === 1)
                    throw Error(invalidEncoding);
                return offset - start;
            };
            base642.test = function test(string2) {
                return /^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$/.test(string2);
            };
        },
    });
    // node_modules/@protobufjs/eventemitter/index.js
    var require_eventemitter = __commonJS({
        'node_modules/@protobufjs/eventemitter/index.js'(exports2, module2) {
            'use strict';
            module2.exports = EventEmitter;
            function EventEmitter() {
                this._listeners = {};
            }
            EventEmitter.prototype.on = function on(evt, fn, ctx) {
                (this._listeners[evt] || (this._listeners[evt] = [])).push({
                    fn,
                    ctx: ctx || this,
                });
                return this;
            };
            EventEmitter.prototype.off = function off(evt, fn) {
                if (evt === void 0)
                    this._listeners = {};
                else {
                    if (fn === void 0)
                        this._listeners[evt] = [];
                    else {
                        var listeners = this._listeners[evt];
                        for (var i = 0; i < listeners.length;)
                            if (listeners[i].fn === fn)
                                listeners.splice(i, 1);
                            else
                                ++i;
                    }
                }
                return this;
            };
            EventEmitter.prototype.emit = function emit(evt) {
                var listeners = this._listeners[evt];
                if (listeners) {
                    var args = [], i = 1;
                    for (; i < arguments.length;)
                        args.push(arguments[i++]);
                    for (i = 0; i < listeners.length;)
                        listeners[i].fn.apply(listeners[i++].ctx, args);
                }
                return this;
            };
        },
    });
    // node_modules/@protobufjs/float/index.js
    var require_float = __commonJS({
        'node_modules/@protobufjs/float/index.js'(exports2, module2) {
            'use strict';
            module2.exports = factory(factory);
            function factory(exports3) {
                if (typeof Float32Array !== 'undefined')
                    (function () {
                        var f32 = new Float32Array([-0]), f8b = new Uint8Array(f32.buffer), le = f8b[3] === 128;
                        function writeFloat_f32_cpy(val, buf, pos) {
                            f32[0] = val;
                            buf[pos] = f8b[0];
                            buf[pos + 1] = f8b[1];
                            buf[pos + 2] = f8b[2];
                            buf[pos + 3] = f8b[3];
                        }
                        function writeFloat_f32_rev(val, buf, pos) {
                            f32[0] = val;
                            buf[pos] = f8b[3];
                            buf[pos + 1] = f8b[2];
                            buf[pos + 2] = f8b[1];
                            buf[pos + 3] = f8b[0];
                        }
                        exports3.writeFloatLE = le
                            ? writeFloat_f32_cpy
                            : writeFloat_f32_rev;
                        exports3.writeFloatBE = le
                            ? writeFloat_f32_rev
                            : writeFloat_f32_cpy;
                        function readFloat_f32_cpy(buf, pos) {
                            f8b[0] = buf[pos];
                            f8b[1] = buf[pos + 1];
                            f8b[2] = buf[pos + 2];
                            f8b[3] = buf[pos + 3];
                            return f32[0];
                        }
                        function readFloat_f32_rev(buf, pos) {
                            f8b[3] = buf[pos];
                            f8b[2] = buf[pos + 1];
                            f8b[1] = buf[pos + 2];
                            f8b[0] = buf[pos + 3];
                            return f32[0];
                        }
                        exports3.readFloatLE = le ? readFloat_f32_cpy : readFloat_f32_rev;
                        exports3.readFloatBE = le ? readFloat_f32_rev : readFloat_f32_cpy;
                    })();
                else
                    (function () {
                        function writeFloat_ieee754(writeUint, val, buf, pos) {
                            var sign = val < 0 ? 1 : 0;
                            if (sign)
                                val = -val;
                            if (val === 0)
                                writeUint(1 / val > 0
                                    ? /* positive */
                                        0
                                    : /* negative 0 */
                                        2147483648, buf, pos);
                            else if (isNaN(val))
                                writeUint(2143289344, buf, pos);
                            else if (val > 34028234663852886e22)
                                writeUint(((sign << 31) | 2139095040) >>> 0, buf, pos);
                            else if (val < 11754943508222875e-54)
                                writeUint(((sign << 31) | Math.round(val / 1401298464324817e-60)) >>> 0, buf, pos);
                            else {
                                var exponent = Math.floor(Math.log(val) / Math.LN2), mantissa = Math.round(val * Math.pow(2, -exponent) * 8388608) &
                                    8388607;
                                writeUint(((sign << 31) | ((exponent + 127) << 23) | mantissa) >>> 0, buf, pos);
                            }
                        }
                        exports3.writeFloatLE = writeFloat_ieee754.bind(null, writeUintLE);
                        exports3.writeFloatBE = writeFloat_ieee754.bind(null, writeUintBE);
                        function readFloat_ieee754(readUint, buf, pos) {
                            var uint = readUint(buf, pos), sign = (uint >> 31) * 2 + 1, exponent = (uint >>> 23) & 255, mantissa = uint & 8388607;
                            return exponent === 255
                                ? mantissa
                                    ? NaN
                                    : sign * Infinity
                                : exponent === 0
                                    ? sign * 1401298464324817e-60 * mantissa
                                    : sign * Math.pow(2, exponent - 150) * (mantissa + 8388608);
                        }
                        exports3.readFloatLE = readFloat_ieee754.bind(null, readUintLE);
                        exports3.readFloatBE = readFloat_ieee754.bind(null, readUintBE);
                    })();
                if (typeof Float64Array !== 'undefined')
                    (function () {
                        var f64 = new Float64Array([-0]), f8b = new Uint8Array(f64.buffer), le = f8b[7] === 128;
                        function writeDouble_f64_cpy(val, buf, pos) {
                            f64[0] = val;
                            buf[pos] = f8b[0];
                            buf[pos + 1] = f8b[1];
                            buf[pos + 2] = f8b[2];
                            buf[pos + 3] = f8b[3];
                            buf[pos + 4] = f8b[4];
                            buf[pos + 5] = f8b[5];
                            buf[pos + 6] = f8b[6];
                            buf[pos + 7] = f8b[7];
                        }
                        function writeDouble_f64_rev(val, buf, pos) {
                            f64[0] = val;
                            buf[pos] = f8b[7];
                            buf[pos + 1] = f8b[6];
                            buf[pos + 2] = f8b[5];
                            buf[pos + 3] = f8b[4];
                            buf[pos + 4] = f8b[3];
                            buf[pos + 5] = f8b[2];
                            buf[pos + 6] = f8b[1];
                            buf[pos + 7] = f8b[0];
                        }
                        exports3.writeDoubleLE = le
                            ? writeDouble_f64_cpy
                            : writeDouble_f64_rev;
                        exports3.writeDoubleBE = le
                            ? writeDouble_f64_rev
                            : writeDouble_f64_cpy;
                        function readDouble_f64_cpy(buf, pos) {
                            f8b[0] = buf[pos];
                            f8b[1] = buf[pos + 1];
                            f8b[2] = buf[pos + 2];
                            f8b[3] = buf[pos + 3];
                            f8b[4] = buf[pos + 4];
                            f8b[5] = buf[pos + 5];
                            f8b[6] = buf[pos + 6];
                            f8b[7] = buf[pos + 7];
                            return f64[0];
                        }
                        function readDouble_f64_rev(buf, pos) {
                            f8b[7] = buf[pos];
                            f8b[6] = buf[pos + 1];
                            f8b[5] = buf[pos + 2];
                            f8b[4] = buf[pos + 3];
                            f8b[3] = buf[pos + 4];
                            f8b[2] = buf[pos + 5];
                            f8b[1] = buf[pos + 6];
                            f8b[0] = buf[pos + 7];
                            return f64[0];
                        }
                        exports3.readDoubleLE = le
                            ? readDouble_f64_cpy
                            : readDouble_f64_rev;
                        exports3.readDoubleBE = le
                            ? readDouble_f64_rev
                            : readDouble_f64_cpy;
                    })();
                else
                    (function () {
                        function writeDouble_ieee754(writeUint, off0, off1, val, buf, pos) {
                            var sign = val < 0 ? 1 : 0;
                            if (sign)
                                val = -val;
                            if (val === 0) {
                                writeUint(0, buf, pos + off0);
                                writeUint(1 / val > 0
                                    ? /* positive */
                                        0
                                    : /* negative 0 */
                                        2147483648, buf, pos + off1);
                            }
                            else if (isNaN(val)) {
                                writeUint(0, buf, pos + off0);
                                writeUint(2146959360, buf, pos + off1);
                            }
                            else if (val > 17976931348623157e292) {
                                writeUint(0, buf, pos + off0);
                                writeUint(((sign << 31) | 2146435072) >>> 0, buf, pos + off1);
                            }
                            else {
                                var mantissa;
                                if (val < 22250738585072014e-324) {
                                    mantissa = val / 5e-324;
                                    writeUint(mantissa >>> 0, buf, pos + off0);
                                    writeUint(((sign << 31) | (mantissa / 4294967296)) >>> 0, buf, pos + off1);
                                }
                                else {
                                    var exponent = Math.floor(Math.log(val) / Math.LN2);
                                    if (exponent === 1024)
                                        exponent = 1023;
                                    mantissa = val * Math.pow(2, -exponent);
                                    writeUint((mantissa * 4503599627370496) >>> 0, buf, pos + off0);
                                    writeUint(((sign << 31) |
                                        ((exponent + 1023) << 20) |
                                        ((mantissa * 1048576) & 1048575)) >>>
                                        0, buf, pos + off1);
                                }
                            }
                        }
                        exports3.writeDoubleLE = writeDouble_ieee754.bind(null, writeUintLE, 0, 4);
                        exports3.writeDoubleBE = writeDouble_ieee754.bind(null, writeUintBE, 4, 0);
                        function readDouble_ieee754(readUint, off0, off1, buf, pos) {
                            var lo = readUint(buf, pos + off0), hi = readUint(buf, pos + off1);
                            var sign = (hi >> 31) * 2 + 1, exponent = (hi >>> 20) & 2047, mantissa = 4294967296 * (hi & 1048575) + lo;
                            return exponent === 2047
                                ? mantissa
                                    ? NaN
                                    : sign * Infinity
                                : exponent === 0
                                    ? sign * 5e-324 * mantissa
                                    : sign *
                                        Math.pow(2, exponent - 1075) *
                                        (mantissa + 4503599627370496);
                        }
                        exports3.readDoubleLE = readDouble_ieee754.bind(null, readUintLE, 0, 4);
                        exports3.readDoubleBE = readDouble_ieee754.bind(null, readUintBE, 4, 0);
                    })();
                return exports3;
            }
            function writeUintLE(val, buf, pos) {
                buf[pos] = val & 255;
                buf[pos + 1] = (val >>> 8) & 255;
                buf[pos + 2] = (val >>> 16) & 255;
                buf[pos + 3] = val >>> 24;
            }
            function writeUintBE(val, buf, pos) {
                buf[pos] = val >>> 24;
                buf[pos + 1] = (val >>> 16) & 255;
                buf[pos + 2] = (val >>> 8) & 255;
                buf[pos + 3] = val & 255;
            }
            function readUintLE(buf, pos) {
                return ((buf[pos] |
                    (buf[pos + 1] << 8) |
                    (buf[pos + 2] << 16) |
                    (buf[pos + 3] << 24)) >>>
                    0);
            }
            function readUintBE(buf, pos) {
                return (((buf[pos] << 24) |
                    (buf[pos + 1] << 16) |
                    (buf[pos + 2] << 8) |
                    buf[pos + 3]) >>>
                    0);
            }
        },
    });
    // node_modules/@protobufjs/inquire/index.js
    var require_inquire = __commonJS({
        'node_modules/@protobufjs/inquire/index.js'(exports, module) {
            'use strict';
            module.exports = inquire;
            function inquire(moduleName) {
                try {
                    var mod = eval('quire'.replace(/^/, 're'))(moduleName);
                    if (mod && (mod.length || Object.keys(mod).length))
                        return mod;
                }
                catch (e) { }
                return null;
            }
        },
    });
    // node_modules/@protobufjs/utf8/index.js
    var require_utf8 = __commonJS({
        'node_modules/@protobufjs/utf8/index.js'(exports2) {
            'use strict';
            var utf8 = exports2;
            utf8.length = function utf8_length(string2) {
                var len = 0, c = 0;
                for (var i = 0; i < string2.length; ++i) {
                    c = string2.charCodeAt(i);
                    if (c < 128)
                        len += 1;
                    else if (c < 2048)
                        len += 2;
                    else if ((c & 64512) === 55296 &&
                        (string2.charCodeAt(i + 1) & 64512) === 56320) {
                        ++i;
                        len += 4;
                    }
                    else
                        len += 3;
                }
                return len;
            };
            utf8.read = function utf8_read(buffer, start, end) {
                var len = end - start;
                if (len < 1)
                    return '';
                var parts = null, chunk = [], i = 0, t;
                while (start < end) {
                    t = buffer[start++];
                    if (t < 128)
                        chunk[i++] = t;
                    else if (t > 191 && t < 224)
                        chunk[i++] = ((t & 31) << 6) | (buffer[start++] & 63);
                    else if (t > 239 && t < 365) {
                        t =
                            (((t & 7) << 18) |
                                ((buffer[start++] & 63) << 12) |
                                ((buffer[start++] & 63) << 6) |
                                (buffer[start++] & 63)) -
                                65536;
                        chunk[i++] = 55296 + (t >> 10);
                        chunk[i++] = 56320 + (t & 1023);
                    }
                    else
                        chunk[i++] =
                            ((t & 15) << 12) |
                                ((buffer[start++] & 63) << 6) |
                                (buffer[start++] & 63);
                    if (i > 8191) {
                        (parts || (parts = [])).push(String.fromCharCode.apply(String, chunk));
                        i = 0;
                    }
                }
                if (parts) {
                    if (i)
                        parts.push(String.fromCharCode.apply(String, chunk.slice(0, i)));
                    return parts.join('');
                }
                return String.fromCharCode.apply(String, chunk.slice(0, i));
            };
            utf8.write = function utf8_write(string2, buffer, offset) {
                var start = offset, c1, c2;
                for (var i = 0; i < string2.length; ++i) {
                    c1 = string2.charCodeAt(i);
                    if (c1 < 128) {
                        buffer[offset++] = c1;
                    }
                    else if (c1 < 2048) {
                        buffer[offset++] = (c1 >> 6) | 192;
                        buffer[offset++] = (c1 & 63) | 128;
                    }
                    else if ((c1 & 64512) === 55296 &&
                        ((c2 = string2.charCodeAt(i + 1)) & 64512) === 56320) {
                        c1 = 65536 + ((c1 & 1023) << 10) + (c2 & 1023);
                        ++i;
                        buffer[offset++] = (c1 >> 18) | 240;
                        buffer[offset++] = ((c1 >> 12) & 63) | 128;
                        buffer[offset++] = ((c1 >> 6) & 63) | 128;
                        buffer[offset++] = (c1 & 63) | 128;
                    }
                    else {
                        buffer[offset++] = (c1 >> 12) | 224;
                        buffer[offset++] = ((c1 >> 6) & 63) | 128;
                        buffer[offset++] = (c1 & 63) | 128;
                    }
                }
                return offset - start;
            };
        },
    });
    // node_modules/@protobufjs/pool/index.js
    var require_pool = __commonJS({
        'node_modules/@protobufjs/pool/index.js'(exports2, module2) {
            'use strict';
            module2.exports = pool;
            function pool(alloc2, slice, size) {
                var SIZE = size || 8192;
                var MAX = SIZE >>> 1;
                var slab = null;
                var offset = SIZE;
                return function pool_alloc(size2) {
                    if (size2 < 1 || size2 > MAX)
                        return alloc2(size2);
                    if (offset + size2 > SIZE) {
                        slab = alloc2(SIZE);
                        offset = 0;
                    }
                    var buf = slice.call(slab, offset, (offset += size2));
                    if (offset & 7)
                        offset = (offset | 7) + 1;
                    return buf;
                };
            }
        },
    });
    // node_modules/protobufjs/src/util/longbits.js
    var require_longbits = __commonJS({
        'node_modules/protobufjs/src/util/longbits.js'(exports2, module2) {
            'use strict';
            module2.exports = LongBits;
            var util = require_minimal();
            function LongBits(lo, hi) {
                this.lo = lo >>> 0;
                this.hi = hi >>> 0;
            }
            var zero = (LongBits.zero = new LongBits(0, 0));
            zero.toNumber = function () {
                return 0;
            };
            zero.zzEncode = zero.zzDecode = function () {
                return this;
            };
            zero.length = function () {
                return 1;
            };
            var zeroHash = (LongBits.zeroHash = '\0\0\0\0\0\0\0\0');
            LongBits.fromNumber = function fromNumber(value) {
                if (value === 0)
                    return zero;
                var sign = value < 0;
                if (sign)
                    value = -value;
                var lo = value >>> 0, hi = ((value - lo) / 4294967296) >>> 0;
                if (sign) {
                    hi = ~hi >>> 0;
                    lo = ~lo >>> 0;
                    if (++lo > 4294967295) {
                        lo = 0;
                        if (++hi > 4294967295)
                            hi = 0;
                    }
                }
                return new LongBits(lo, hi);
            };
            LongBits.from = function from3(value) {
                if (typeof value === 'number')
                    return LongBits.fromNumber(value);
                if (util.isString(value)) {
                    if (util.Long)
                        value = util.Long.fromString(value);
                    else
                        return LongBits.fromNumber(parseInt(value, 10));
                }
                return value.low || value.high
                    ? new LongBits(value.low >>> 0, value.high >>> 0)
                    : zero;
            };
            LongBits.prototype.toNumber = function toNumber(unsigned) {
                if (!unsigned && this.hi >>> 31) {
                    var lo = (~this.lo + 1) >>> 0, hi = ~this.hi >>> 0;
                    if (!lo)
                        hi = (hi + 1) >>> 0;
                    return -(lo + hi * 4294967296);
                }
                return this.lo + this.hi * 4294967296;
            };
            LongBits.prototype.toLong = function toLong(unsigned) {
                return util.Long
                    ? new util.Long(this.lo | 0, this.hi | 0, Boolean(unsigned))
                    : {
                        low: this.lo | 0,
                        high: this.hi | 0,
                        unsigned: Boolean(unsigned),
                    };
            };
            var charCodeAt = String.prototype.charCodeAt;
            LongBits.fromHash = function fromHash(hash) {
                if (hash === zeroHash)
                    return zero;
                return new LongBits((charCodeAt.call(hash, 0) |
                    (charCodeAt.call(hash, 1) << 8) |
                    (charCodeAt.call(hash, 2) << 16) |
                    (charCodeAt.call(hash, 3) << 24)) >>>
                    0, (charCodeAt.call(hash, 4) |
                    (charCodeAt.call(hash, 5) << 8) |
                    (charCodeAt.call(hash, 6) << 16) |
                    (charCodeAt.call(hash, 7) << 24)) >>>
                    0);
            };
            LongBits.prototype.toHash = function toHash() {
                return String.fromCharCode(this.lo & 255, (this.lo >>> 8) & 255, (this.lo >>> 16) & 255, this.lo >>> 24, this.hi & 255, (this.hi >>> 8) & 255, (this.hi >>> 16) & 255, this.hi >>> 24);
            };
            LongBits.prototype.zzEncode = function zzEncode() {
                var mask = this.hi >> 31;
                this.hi = (((this.hi << 1) | (this.lo >>> 31)) ^ mask) >>> 0;
                this.lo = ((this.lo << 1) ^ mask) >>> 0;
                return this;
            };
            LongBits.prototype.zzDecode = function zzDecode() {
                var mask = -(this.lo & 1);
                this.lo = (((this.lo >>> 1) | (this.hi << 31)) ^ mask) >>> 0;
                this.hi = ((this.hi >>> 1) ^ mask) >>> 0;
                return this;
            };
            LongBits.prototype.length = function length2() {
                var part0 = this.lo, part1 = ((this.lo >>> 28) | (this.hi << 4)) >>> 0, part2 = this.hi >>> 24;
                return part2 === 0
                    ? part1 === 0
                        ? part0 < 16384
                            ? part0 < 128
                                ? 1
                                : 2
                            : part0 < 2097152
                                ? 3
                                : 4
                        : part1 < 16384
                            ? part1 < 128
                                ? 5
                                : 6
                            : part1 < 2097152
                                ? 7
                                : 8
                    : part2 < 128
                        ? 9
                        : 10;
            };
        },
    });
    // node_modules/protobufjs/src/util/minimal.js
    var require_minimal = __commonJS({
        'node_modules/protobufjs/src/util/minimal.js'(exports2) {
            'use strict';
            var util = exports2;
            util.asPromise = require_aspromise();
            util.base64 = require_base64();
            util.EventEmitter = require_eventemitter();
            util.float = require_float();
            util.inquire = require_inquire();
            util.utf8 = require_utf8();
            util.pool = require_pool();
            util.LongBits = require_longbits();
            util.isNode = Boolean(typeof global !== 'undefined' &&
                global &&
                global.process &&
                global.process.versions &&
                global.process.versions.node);
            util.global =
                (util.isNode && global) ||
                    (typeof window !== 'undefined' && window) ||
                    (typeof self !== 'undefined' && self) ||
                    exports2;
            util.emptyArray = Object.freeze
                ? Object.freeze([])
                : /* istanbul ignore next */
                    [];
            util.emptyObject = Object.freeze
                ? Object.freeze({})
                : /* istanbul ignore next */
                    {};
            util.isInteger =
                Number.isInteger /* istanbul ignore next */ ||
                    function isInteger(value) {
                        return (typeof value === 'number' &&
                            isFinite(value) &&
                            Math.floor(value) === value);
                    };
            util.isString = function isString(value) {
                return typeof value === 'string' || value instanceof String;
            };
            util.isObject = function isObject(value) {
                return value && typeof value === 'object';
            };
            util.isset =
                /**
                 * Checks if a property on a message is considered to be present.
                 * @param {Object} obj Plain object or message instance
                 * @param {string} prop Property name
                 * @returns {boolean} `true` if considered to be present, otherwise `false`
                 */
                util.isSet = function isSet(obj, prop) {
                    var value = obj[prop];
                    if (value != null && obj.hasOwnProperty(prop))
                        return (typeof value !== 'object' ||
                            (Array.isArray(value)
                                ? value.length
                                : Object.keys(value).length) > 0);
                    return false;
                };
            util.Buffer = (function () {
                try {
                    var Buffer2 = util.inquire('buffer').Buffer;
                    return Buffer2.prototype.utf8Write
                        ? Buffer2
                        : /* istanbul ignore next */
                            null;
                }
                catch (e) {
                    return null;
                }
            })();
            util._Buffer_from = null;
            util._Buffer_allocUnsafe = null;
            util.newBuffer = function newBuffer(sizeOrArray) {
                return typeof sizeOrArray === 'number'
                    ? util.Buffer
                        ? util._Buffer_allocUnsafe(sizeOrArray)
                        : new util.Array(sizeOrArray)
                    : util.Buffer
                        ? util._Buffer_from(sizeOrArray)
                        : typeof Uint8Array === 'undefined'
                            ? sizeOrArray
                            : new Uint8Array(sizeOrArray);
            };
            util.Array = typeof Uint8Array !== 'undefined' ? Uint8Array : Array;
            util.Long =
                /* istanbul ignore next */
                (util.global.dcodeIO /* istanbul ignore next */ &&
                    util.global.dcodeIO.Long) /* istanbul ignore next */ ||
                    util.global.Long ||
                    util.inquire('long');
            util.key2Re = /^true|false|0|1$/;
            util.key32Re = /^-?(?:0|[1-9][0-9]*)$/;
            util.key64Re = /^(?:[\\x00-\\xff]{8}|-?(?:0|[1-9][0-9]*))$/;
            util.longToHash = function longToHash(value) {
                return value
                    ? util.LongBits.from(value).toHash()
                    : util.LongBits.zeroHash;
            };
            util.longFromHash = function longFromHash(hash, unsigned) {
                var bits = util.LongBits.fromHash(hash);
                if (util.Long)
                    return util.Long.fromBits(bits.lo, bits.hi, unsigned);
                return bits.toNumber(Boolean(unsigned));
            };
            function merge(dst, src2, ifNotSet) {
                for (var keys = Object.keys(src2), i = 0; i < keys.length; ++i)
                    if (dst[keys[i]] === void 0 || !ifNotSet)
                        dst[keys[i]] = src2[keys[i]];
                return dst;
            }
            util.merge = merge;
            util.lcFirst = function lcFirst(str) {
                return str.charAt(0).toLowerCase() + str.substring(1);
            };
            function newError(name4) {
                function CustomError(message, properties) {
                    if (!(this instanceof CustomError))
                        return new CustomError(message, properties);
                    Object.defineProperty(this, 'message', {
                        get: function () {
                            return message;
                        },
                    });
                    if (Error.captureStackTrace)
                        Error.captureStackTrace(this, CustomError);
                    else
                        Object.defineProperty(this, 'stack', {
                            value: new Error().stack || '',
                        });
                    if (properties)
                        merge(this, properties);
                }
                CustomError.prototype = Object.create(Error.prototype, {
                    constructor: {
                        value: CustomError,
                        writable: true,
                        enumerable: false,
                        configurable: true,
                    },
                    name: {
                        get() {
                            return name4;
                        },
                        set: void 0,
                        enumerable: false,
                        // configurable: false would accurately preserve the behavior of
                        // the original, but I'm guessing that was not intentional.
                        // For an actual error subclass, this property would
                        // be configurable.
                        configurable: true,
                    },
                    toString: {
                        value() {
                            return this.name + ': ' + this.message;
                        },
                        writable: true,
                        enumerable: false,
                        configurable: true,
                    },
                });
                return CustomError;
            }
            util.newError = newError;
            util.ProtocolError = newError('ProtocolError');
            util.oneOfGetter = function getOneOf(fieldNames) {
                var fieldMap = {};
                for (var i = 0; i < fieldNames.length; ++i)
                    fieldMap[fieldNames[i]] = 1;
                return function () {
                    for (var keys = Object.keys(this), i2 = keys.length - 1; i2 > -1; --i2)
                        if (fieldMap[keys[i2]] === 1 &&
                            this[keys[i2]] !== void 0 &&
                            this[keys[i2]] !== null)
                            return keys[i2];
                };
            };
            util.oneOfSetter = function setOneOf(fieldNames) {
                return function (name4) {
                    for (var i = 0; i < fieldNames.length; ++i)
                        if (fieldNames[i] !== name4)
                            delete this[fieldNames[i]];
                };
            };
            util.toJSONOptions = {
                longs: String,
                enums: String,
                bytes: String,
                json: true,
            };
            util._configure = function () {
                var Buffer2 = util.Buffer;
                if (!Buffer2) {
                    util._Buffer_from = util._Buffer_allocUnsafe = null;
                    return;
                }
                util._Buffer_from =
                    (Buffer2.from !== Uint8Array.from &&
                        Buffer2.from) /* istanbul ignore next */ ||
                        function Buffer_from(value, encoding) {
                            return new Buffer2(value, encoding);
                        };
                util._Buffer_allocUnsafe =
                    Buffer2.allocUnsafe /* istanbul ignore next */ ||
                        function Buffer_allocUnsafe(size) {
                            return new Buffer2(size);
                        };
            };
        },
    });
    // node_modules/protobufjs/src/writer.js
    var require_writer = __commonJS({
        'node_modules/protobufjs/src/writer.js'(exports2, module2) {
            'use strict';
            module2.exports = Writer;
            var util = require_minimal();
            var BufferWriter;
            var LongBits = util.LongBits;
            var base642 = util.base64;
            var utf8 = util.utf8;
            function Op(fn, len, val) {
                this.fn = fn;
                this.len = len;
                this.next = void 0;
                this.val = val;
            }
            function noop() { }
            function State(writer) {
                this.head = writer.head;
                this.tail = writer.tail;
                this.len = writer.len;
                this.next = writer.states;
            }
            function Writer() {
                this.len = 0;
                this.head = new Op(noop, 0, 0);
                this.tail = this.head;
                this.states = null;
            }
            var create3 = function create4() {
                return util.Buffer
                    ? function create_buffer_setup() {
                        return (Writer.create = function create_buffer() {
                            return new BufferWriter();
                        })();
                    }
                    : function create_array() {
                        return new Writer();
                    };
            };
            Writer.create = create3();
            Writer.alloc = function alloc2(size) {
                return new util.Array(size);
            };
            if (util.Array !== Array)
                Writer.alloc = util.pool(Writer.alloc, util.Array.prototype.subarray);
            Writer.prototype._push = function push(fn, len, val) {
                this.tail = this.tail.next = new Op(fn, len, val);
                this.len += len;
                return this;
            };
            function writeByte(val, buf, pos) {
                buf[pos] = val & 255;
            }
            function writeVarint32(val, buf, pos) {
                while (val > 127) {
                    buf[pos++] = (val & 127) | 128;
                    val >>>= 7;
                }
                buf[pos] = val;
            }
            function VarintOp(len, val) {
                this.len = len;
                this.next = void 0;
                this.val = val;
            }
            VarintOp.prototype = Object.create(Op.prototype);
            VarintOp.prototype.fn = writeVarint32;
            Writer.prototype.uint32 = function write_uint32(value) {
                this.len += (this.tail = this.tail.next =
                    new VarintOp((value = value >>> 0) < 128
                        ? 1
                        : value < 16384
                            ? 2
                            : value < 2097152
                                ? 3
                                : value < 268435456
                                    ? 4
                                    : 5, value)).len;
                return this;
            };
            Writer.prototype.int32 = function write_int32(value) {
                return value < 0
                    ? this._push(writeVarint64, 10, LongBits.fromNumber(value))
                    : this.uint32(value);
            };
            Writer.prototype.sint32 = function write_sint32(value) {
                return this.uint32(((value << 1) ^ (value >> 31)) >>> 0);
            };
            function writeVarint64(val, buf, pos) {
                while (val.hi) {
                    buf[pos++] = (val.lo & 127) | 128;
                    val.lo = ((val.lo >>> 7) | (val.hi << 25)) >>> 0;
                    val.hi >>>= 7;
                }
                while (val.lo > 127) {
                    buf[pos++] = (val.lo & 127) | 128;
                    val.lo = val.lo >>> 7;
                }
                buf[pos++] = val.lo;
            }
            Writer.prototype.uint64 = function write_uint64(value) {
                var bits = LongBits.from(value);
                return this._push(writeVarint64, bits.length(), bits);
            };
            Writer.prototype.int64 = Writer.prototype.uint64;
            Writer.prototype.sint64 = function write_sint64(value) {
                var bits = LongBits.from(value).zzEncode();
                return this._push(writeVarint64, bits.length(), bits);
            };
            Writer.prototype.bool = function write_bool(value) {
                return this._push(writeByte, 1, value ? 1 : 0);
            };
            function writeFixed32(val, buf, pos) {
                buf[pos] = val & 255;
                buf[pos + 1] = (val >>> 8) & 255;
                buf[pos + 2] = (val >>> 16) & 255;
                buf[pos + 3] = val >>> 24;
            }
            Writer.prototype.fixed32 = function write_fixed32(value) {
                return this._push(writeFixed32, 4, value >>> 0);
            };
            Writer.prototype.sfixed32 = Writer.prototype.fixed32;
            Writer.prototype.fixed64 = function write_fixed64(value) {
                var bits = LongBits.from(value);
                return this._push(writeFixed32, 4, bits.lo)._push(writeFixed32, 4, bits.hi);
            };
            Writer.prototype.sfixed64 = Writer.prototype.fixed64;
            Writer.prototype.float = function write_float(value) {
                return this._push(util.float.writeFloatLE, 4, value);
            };
            Writer.prototype.double = function write_double(value) {
                return this._push(util.float.writeDoubleLE, 8, value);
            };
            var writeBytes = util.Array.prototype.set
                ? function writeBytes_set(val, buf, pos) {
                    buf.set(val, pos);
                }
                : function writeBytes_for(val, buf, pos) {
                    for (var i = 0; i < val.length; ++i)
                        buf[pos + i] = val[i];
                };
            Writer.prototype.bytes = function write_bytes(value) {
                var len = value.length >>> 0;
                if (!len)
                    return this._push(writeByte, 1, 0);
                if (util.isString(value)) {
                    var buf = Writer.alloc((len = base642.length(value)));
                    base642.decode(value, buf, 0);
                    value = buf;
                }
                return this.uint32(len)._push(writeBytes, len, value);
            };
            Writer.prototype.string = function write_string(value) {
                var len = utf8.length(value);
                return len
                    ? this.uint32(len)._push(utf8.write, len, value)
                    : this._push(writeByte, 1, 0);
            };
            Writer.prototype.fork = function fork() {
                this.states = new State(this);
                this.head = this.tail = new Op(noop, 0, 0);
                this.len = 0;
                return this;
            };
            Writer.prototype.reset = function reset() {
                if (this.states) {
                    this.head = this.states.head;
                    this.tail = this.states.tail;
                    this.len = this.states.len;
                    this.states = this.states.next;
                }
                else {
                    this.head = this.tail = new Op(noop, 0, 0);
                    this.len = 0;
                }
                return this;
            };
            Writer.prototype.ldelim = function ldelim() {
                var head = this.head, tail = this.tail, len = this.len;
                this.reset().uint32(len);
                if (len) {
                    this.tail.next = head.next;
                    this.tail = tail;
                    this.len += len;
                }
                return this;
            };
            Writer.prototype.finish = function finish() {
                var head = this.head.next, buf = this.constructor.alloc(this.len), pos = 0;
                while (head) {
                    head.fn(head.val, buf, pos);
                    pos += head.len;
                    head = head.next;
                }
                return buf;
            };
            Writer._configure = function (BufferWriter_) {
                BufferWriter = BufferWriter_;
                Writer.create = create3();
                BufferWriter._configure();
            };
        },
    });
    // node_modules/protobufjs/src/writer_buffer.js
    var require_writer_buffer = __commonJS({
        'node_modules/protobufjs/src/writer_buffer.js'(exports2, module2) {
            'use strict';
            module2.exports = BufferWriter;
            var Writer = require_writer();
            (BufferWriter.prototype = Object.create(Writer.prototype)).constructor =
                BufferWriter;
            var util = require_minimal();
            function BufferWriter() {
                Writer.call(this);
            }
            BufferWriter._configure = function () {
                BufferWriter.alloc = util._Buffer_allocUnsafe;
                BufferWriter.writeBytesBuffer =
                    util.Buffer &&
                        util.Buffer.prototype instanceof Uint8Array &&
                        util.Buffer.prototype.set.name === 'set'
                        ? function writeBytesBuffer_set(val, buf, pos) {
                            buf.set(val, pos);
                        }
                        : function writeBytesBuffer_copy(val, buf, pos) {
                            if (val.copy)
                                val.copy(buf, pos, 0, val.length);
                            else
                                for (var i = 0; i < val.length;)
                                    buf[pos++] = val[i++];
                        };
            };
            BufferWriter.prototype.bytes = function write_bytes_buffer(value) {
                if (util.isString(value))
                    value = util._Buffer_from(value, 'base64');
                var len = value.length >>> 0;
                this.uint32(len);
                if (len)
                    this._push(BufferWriter.writeBytesBuffer, len, value);
                return this;
            };
            function writeStringBuffer(val, buf, pos) {
                if (val.length < 40)
                    util.utf8.write(val, buf, pos);
                else if (buf.utf8Write)
                    buf.utf8Write(val, pos);
                else
                    buf.write(val, pos);
            }
            BufferWriter.prototype.string = function write_string_buffer(value) {
                var len = util.Buffer.byteLength(value);
                this.uint32(len);
                if (len)
                    this._push(writeStringBuffer, len, value);
                return this;
            };
            BufferWriter._configure();
        },
    });
    // node_modules/protobufjs/src/reader.js
    var require_reader = __commonJS({
        'node_modules/protobufjs/src/reader.js'(exports2, module2) {
            'use strict';
            module2.exports = Reader;
            var util = require_minimal();
            var BufferReader;
            var LongBits = util.LongBits;
            var utf8 = util.utf8;
            function indexOutOfRange(reader, writeLength) {
                return RangeError('index out of range: ' +
                    reader.pos +
                    ' + ' +
                    (writeLength || 1) +
                    ' > ' +
                    reader.len);
            }
            function Reader(buffer) {
                this.buf = buffer;
                this.pos = 0;
                this.len = buffer.length;
            }
            var create_array = typeof Uint8Array !== 'undefined'
                ? function create_typed_array(buffer) {
                    if (buffer instanceof Uint8Array || Array.isArray(buffer))
                        return new Reader(buffer);
                    throw Error('illegal buffer');
                }
                : function create_array2(buffer) {
                    if (Array.isArray(buffer))
                        return new Reader(buffer);
                    throw Error('illegal buffer');
                };
            var create3 = function create4() {
                return util.Buffer
                    ? function create_buffer_setup(buffer) {
                        return (Reader.create = function create_buffer(buffer2) {
                            return util.Buffer.isBuffer(buffer2)
                                ? new BufferReader(buffer2)
                                : create_array(buffer2);
                        })(buffer);
                    }
                    : create_array;
            };
            Reader.create = create3();
            Reader.prototype._slice =
                util.Array.prototype.subarray /* istanbul ignore next */ ||
                    util.Array.prototype.slice;
            Reader.prototype.uint32 = (function read_uint32_setup() {
                var value = 4294967295;
                return function read_uint32() {
                    value = (this.buf[this.pos] & 127) >>> 0;
                    if (this.buf[this.pos++] < 128)
                        return value;
                    value = (value | ((this.buf[this.pos] & 127) << 7)) >>> 0;
                    if (this.buf[this.pos++] < 128)
                        return value;
                    value = (value | ((this.buf[this.pos] & 127) << 14)) >>> 0;
                    if (this.buf[this.pos++] < 128)
                        return value;
                    value = (value | ((this.buf[this.pos] & 127) << 21)) >>> 0;
                    if (this.buf[this.pos++] < 128)
                        return value;
                    value = (value | ((this.buf[this.pos] & 15) << 28)) >>> 0;
                    if (this.buf[this.pos++] < 128)
                        return value;
                    if ((this.pos += 5) > this.len) {
                        this.pos = this.len;
                        throw indexOutOfRange(this, 10);
                    }
                    return value;
                };
            })();
            Reader.prototype.int32 = function read_int32() {
                return this.uint32() | 0;
            };
            Reader.prototype.sint32 = function read_sint32() {
                var value = this.uint32();
                return ((value >>> 1) ^ -(value & 1)) | 0;
            };
            function readLongVarint() {
                var bits = new LongBits(0, 0);
                var i = 0;
                if (this.len - this.pos > 4) {
                    for (; i < 4; ++i) {
                        bits.lo = (bits.lo | ((this.buf[this.pos] & 127) << (i * 7))) >>> 0;
                        if (this.buf[this.pos++] < 128)
                            return bits;
                    }
                    bits.lo = (bits.lo | ((this.buf[this.pos] & 127) << 28)) >>> 0;
                    bits.hi = (bits.hi | ((this.buf[this.pos] & 127) >> 4)) >>> 0;
                    if (this.buf[this.pos++] < 128)
                        return bits;
                    i = 0;
                }
                else {
                    for (; i < 3; ++i) {
                        if (this.pos >= this.len)
                            throw indexOutOfRange(this);
                        bits.lo = (bits.lo | ((this.buf[this.pos] & 127) << (i * 7))) >>> 0;
                        if (this.buf[this.pos++] < 128)
                            return bits;
                    }
                    bits.lo = (bits.lo | ((this.buf[this.pos++] & 127) << (i * 7))) >>> 0;
                    return bits;
                }
                if (this.len - this.pos > 4) {
                    for (; i < 5; ++i) {
                        bits.hi =
                            (bits.hi | ((this.buf[this.pos] & 127) << (i * 7 + 3))) >>> 0;
                        if (this.buf[this.pos++] < 128)
                            return bits;
                    }
                }
                else {
                    for (; i < 5; ++i) {
                        if (this.pos >= this.len)
                            throw indexOutOfRange(this);
                        bits.hi =
                            (bits.hi | ((this.buf[this.pos] & 127) << (i * 7 + 3))) >>> 0;
                        if (this.buf[this.pos++] < 128)
                            return bits;
                    }
                }
                throw Error('invalid varint encoding');
            }
            Reader.prototype.bool = function read_bool() {
                return this.uint32() !== 0;
            };
            function readFixed32_end(buf, end) {
                return ((buf[end - 4] |
                    (buf[end - 3] << 8) |
                    (buf[end - 2] << 16) |
                    (buf[end - 1] << 24)) >>>
                    0);
            }
            Reader.prototype.fixed32 = function read_fixed32() {
                if (this.pos + 4 > this.len)
                    throw indexOutOfRange(this, 4);
                return readFixed32_end(this.buf, (this.pos += 4));
            };
            Reader.prototype.sfixed32 = function read_sfixed32() {
                if (this.pos + 4 > this.len)
                    throw indexOutOfRange(this, 4);
                return readFixed32_end(this.buf, (this.pos += 4)) | 0;
            };
            function readFixed64() {
                if (this.pos + 8 > this.len)
                    throw indexOutOfRange(this, 8);
                return new LongBits(readFixed32_end(this.buf, (this.pos += 4)), readFixed32_end(this.buf, (this.pos += 4)));
            }
            Reader.prototype.float = function read_float() {
                if (this.pos + 4 > this.len)
                    throw indexOutOfRange(this, 4);
                var value = util.float.readFloatLE(this.buf, this.pos);
                this.pos += 4;
                return value;
            };
            Reader.prototype.double = function read_double() {
                if (this.pos + 8 > this.len)
                    throw indexOutOfRange(this, 4);
                var value = util.float.readDoubleLE(this.buf, this.pos);
                this.pos += 8;
                return value;
            };
            Reader.prototype.bytes = function read_bytes() {
                var length2 = this.uint32(), start = this.pos, end = this.pos + length2;
                if (end > this.len)
                    throw indexOutOfRange(this, length2);
                this.pos += length2;
                if (Array.isArray(this.buf))
                    return this.buf.slice(start, end);
                return start === end
                    ? new this.buf.constructor(0)
                    : this._slice.call(this.buf, start, end);
            };
            Reader.prototype.string = function read_string() {
                var bytes = this.bytes();
                return utf8.read(bytes, 0, bytes.length);
            };
            Reader.prototype.skip = function skip(length2) {
                if (typeof length2 === 'number') {
                    if (this.pos + length2 > this.len)
                        throw indexOutOfRange(this, length2);
                    this.pos += length2;
                }
                else {
                    do {
                        if (this.pos >= this.len)
                            throw indexOutOfRange(this);
                    } while (this.buf[this.pos++] & 128);
                }
                return this;
            };
            Reader.prototype.skipType = function (wireType) {
                switch (wireType) {
                    case 0:
                        this.skip();
                        break;
                    case 1:
                        this.skip(8);
                        break;
                    case 2:
                        this.skip(this.uint32());
                        break;
                    case 3:
                        while ((wireType = this.uint32() & 7) !== 4) {
                            this.skipType(wireType);
                        }
                        break;
                    case 5:
                        this.skip(4);
                        break;
                    default:
                        throw Error('invalid wire type ' + wireType + ' at offset ' + this.pos);
                }
                return this;
            };
            Reader._configure = function (BufferReader_) {
                BufferReader = BufferReader_;
                Reader.create = create3();
                BufferReader._configure();
                var fn = util.Long
                    ? 'toLong'
                    : /* istanbul ignore next */
                        'toNumber';
                util.merge(Reader.prototype, {
                    int64: function read_int64() {
                        return readLongVarint.call(this)[fn](false);
                    },
                    uint64: function read_uint64() {
                        return readLongVarint.call(this)[fn](true);
                    },
                    sint64: function read_sint64() {
                        return readLongVarint.call(this).zzDecode()[fn](false);
                    },
                    fixed64: function read_fixed64() {
                        return readFixed64.call(this)[fn](true);
                    },
                    sfixed64: function read_sfixed64() {
                        return readFixed64.call(this)[fn](false);
                    },
                });
            };
        },
    });
    // node_modules/protobufjs/src/reader_buffer.js
    var require_reader_buffer = __commonJS({
        'node_modules/protobufjs/src/reader_buffer.js'(exports2, module2) {
            'use strict';
            module2.exports = BufferReader;
            var Reader = require_reader();
            (BufferReader.prototype = Object.create(Reader.prototype)).constructor =
                BufferReader;
            var util = require_minimal();
            function BufferReader(buffer) {
                Reader.call(this, buffer);
            }
            BufferReader._configure = function () {
                if (util.Buffer)
                    BufferReader.prototype._slice = util.Buffer.prototype.slice;
            };
            BufferReader.prototype.string = function read_string_buffer() {
                var len = this.uint32();
                return this.buf.utf8Slice
                    ? this.buf.utf8Slice(this.pos, (this.pos = Math.min(this.pos + len, this.len)))
                    : this.buf.toString('utf-8', this.pos, (this.pos = Math.min(this.pos + len, this.len)));
            };
            BufferReader._configure();
        },
    });
    // node_modules/protobufjs/src/rpc/service.js
    var require_service = __commonJS({
        'node_modules/protobufjs/src/rpc/service.js'(exports2, module2) {
            'use strict';
            module2.exports = Service;
            var util = require_minimal();
            (Service.prototype = Object.create(util.EventEmitter.prototype)).constructor = Service;
            function Service(rpcImpl, requestDelimited, responseDelimited) {
                if (typeof rpcImpl !== 'function')
                    throw TypeError('rpcImpl must be a function');
                util.EventEmitter.call(this);
                this.rpcImpl = rpcImpl;
                this.requestDelimited = Boolean(requestDelimited);
                this.responseDelimited = Boolean(responseDelimited);
            }
            Service.prototype.rpcCall = function rpcCall(method, requestCtor, responseCtor, request, callback) {
                if (!request)
                    throw TypeError('request must be specified');
                var self2 = this;
                if (!callback)
                    return util.asPromise(rpcCall, self2, method, requestCtor, responseCtor, request);
                if (!self2.rpcImpl) {
                    setTimeout(function () {
                        callback(Error('already ended'));
                    }, 0);
                    return void 0;
                }
                try {
                    return self2.rpcImpl(method, requestCtor[self2.requestDelimited ? 'encodeDelimited' : 'encode'](request).finish(), function rpcCallback(err, response) {
                        if (err) {
                            self2.emit('error', err, method);
                            return callback(err);
                        }
                        if (response === null) {
                            self2.end(
                            /* endedByRPC */
                            true);
                            return void 0;
                        }
                        if (!(response instanceof responseCtor)) {
                            try {
                                response =
                                    responseCtor[self2.responseDelimited ? 'decodeDelimited' : 'decode'](response);
                            }
                            catch (err2) {
                                self2.emit('error', err2, method);
                                return callback(err2);
                            }
                        }
                        self2.emit('data', response, method);
                        return callback(null, response);
                    });
                }
                catch (err) {
                    self2.emit('error', err, method);
                    setTimeout(function () {
                        callback(err);
                    }, 0);
                    return void 0;
                }
            };
            Service.prototype.end = function end(endedByRPC) {
                if (this.rpcImpl) {
                    if (!endedByRPC)
                        this.rpcImpl(null, null, null);
                    this.rpcImpl = null;
                    this.emit('end').off();
                }
                return this;
            };
        },
    });
    // node_modules/protobufjs/src/rpc.js
    var require_rpc = __commonJS({
        'node_modules/protobufjs/src/rpc.js'(exports2) {
            'use strict';
            var rpc = exports2;
            rpc.Service = require_service();
        },
    });
    // node_modules/protobufjs/src/roots.js
    var require_roots = __commonJS({
        'node_modules/protobufjs/src/roots.js'(exports2, module2) {
            'use strict';
            module2.exports = {};
        },
    });
    // node_modules/protobufjs/src/index-minimal.js
    var require_index_minimal = __commonJS({
        'node_modules/protobufjs/src/index-minimal.js'(exports2) {
            'use strict';
            var protobuf = exports2;
            protobuf.build = 'minimal';
            protobuf.Writer = require_writer();
            protobuf.BufferWriter = require_writer_buffer();
            protobuf.Reader = require_reader();
            protobuf.BufferReader = require_reader_buffer();
            protobuf.util = require_minimal();
            protobuf.rpc = require_rpc();
            protobuf.roots = require_roots();
            protobuf.configure = configure;
            function configure() {
                protobuf.util._configure();
                protobuf.Writer._configure(protobuf.BufferWriter);
                protobuf.Reader._configure(protobuf.BufferReader);
            }
            configure();
        },
    });
    // node_modules/protobufjs/minimal.js
    var require_minimal2 = __commonJS({
        'node_modules/protobufjs/minimal.js'(exports2, module2) {
            'use strict';
            module2.exports = require_index_minimal();
        },
    });
    // node_modules/rabin-wasm/src/rabin.js
    var require_rabin = __commonJS({
        'node_modules/rabin-wasm/src/rabin.js'(exports2, module2) {
            var Rabin = class {
                /**
                 * Creates an instance of Rabin.
                 * @param { import("./../dist/rabin-wasm") } asModule
                 * @param {number} [bits=12]
                 * @param {number} [min=8 * 1024]
                 * @param {number} [max=32 * 1024]
                 * @param {number} polynomial
                 * @memberof Rabin
                 */
                constructor(asModule, bits = 12, min = 8 * 1024, max = 32 * 1024, windowSize = 64, polynomial) {
                    this.bits = bits;
                    this.min = min;
                    this.max = max;
                    this.asModule = asModule;
                    this.rabin = new asModule.Rabin(bits, min, max, windowSize, polynomial);
                    this.polynomial = polynomial;
                }
                /**
                 * Fingerprints the buffer
                 *
                 * @param {Uint8Array} buf
                 * @returns {Array<number>}
                 * @memberof Rabin
                 */
                fingerprint(buf) {
                    const { __retain, __release, __allocArray, __getInt32Array, Int32Array_ID, Uint8Array_ID, } = this.asModule;
                    const lengths = new Int32Array(Math.ceil(buf.length / this.min));
                    const lengthsPtr = __retain(__allocArray(Int32Array_ID, lengths));
                    const pointer = __retain(__allocArray(Uint8Array_ID, buf));
                    const out = this.rabin.fingerprint(pointer, lengthsPtr);
                    const processed = __getInt32Array(out);
                    __release(pointer);
                    __release(lengthsPtr);
                    const end = processed.indexOf(0);
                    return end >= 0 ? processed.subarray(0, end) : processed;
                }
            };
            module2.exports = Rabin;
        },
    });
    // node_modules/@assemblyscript/loader/index.js
    var require_loader = __commonJS({
        'node_modules/@assemblyscript/loader/index.js'(exports2) {
            'use strict';
            var ID_OFFSET = -8;
            var SIZE_OFFSET = -4;
            var ARRAYBUFFER_ID = 0;
            var STRING_ID = 1;
            var ARRAYBUFFERVIEW = 1 << 0;
            var ARRAY = 1 << 1;
            var SET = 1 << 2;
            var MAP = 1 << 3;
            var VAL_ALIGN_OFFSET = 5;
            var VAL_ALIGN = 1 << VAL_ALIGN_OFFSET;
            var VAL_SIGNED = 1 << 10;
            var VAL_FLOAT = 1 << 11;
            var VAL_NULLABLE = 1 << 12;
            var VAL_MANAGED = 1 << 13;
            var KEY_ALIGN_OFFSET = 14;
            var KEY_ALIGN = 1 << KEY_ALIGN_OFFSET;
            var KEY_SIGNED = 1 << 19;
            var KEY_FLOAT = 1 << 20;
            var KEY_NULLABLE = 1 << 21;
            var KEY_MANAGED = 1 << 22;
            var ARRAYBUFFERVIEW_BUFFER_OFFSET = 0;
            var ARRAYBUFFERVIEW_DATASTART_OFFSET = 4;
            var ARRAYBUFFERVIEW_DATALENGTH_OFFSET = 8;
            var ARRAYBUFFERVIEW_SIZE = 12;
            var ARRAY_LENGTH_OFFSET = 12;
            var ARRAY_SIZE = 16;
            var BIGINT = typeof BigUint64Array !== 'undefined';
            var THIS = Symbol();
            var CHUNKSIZE = 1024;
            function getStringImpl(buffer, ptr) {
                const U32 = new Uint32Array(buffer);
                const U16 = new Uint16Array(buffer);
                var length2 = U32[(ptr + SIZE_OFFSET) >>> 2] >>> 1;
                var offset = ptr >>> 1;
                if (length2 <= CHUNKSIZE)
                    return String.fromCharCode.apply(String, U16.subarray(offset, offset + length2));
                const parts = [];
                do {
                    const last = U16[offset + CHUNKSIZE - 1];
                    const size = last >= 55296 && last < 56320 ? CHUNKSIZE - 1 : CHUNKSIZE;
                    parts.push(String.fromCharCode.apply(String, U16.subarray(offset, (offset += size))));
                    length2 -= size;
                } while (length2 > CHUNKSIZE);
                return (parts.join('') +
                    String.fromCharCode.apply(String, U16.subarray(offset, offset + length2)));
            }
            function preInstantiate(imports) {
                const baseModule = {};
                function getString(memory, ptr) {
                    if (!memory)
                        return '<yet unknown>';
                    return getStringImpl(memory.buffer, ptr);
                }
                const env = (imports.env = imports.env || {});
                env.abort =
                    env.abort ||
                        function abort(mesg, file, line, colm) {
                            const memory = baseModule.memory || env.memory;
                            throw Error('abort: ' +
                                getString(memory, mesg) +
                                ' at ' +
                                getString(memory, file) +
                                ':' +
                                line +
                                ':' +
                                colm);
                        };
                env.trace =
                    env.trace ||
                        function trace(mesg, n) {
                            const memory = baseModule.memory || env.memory;
                            console.log('trace: ' +
                                getString(memory, mesg) +
                                (n ? ' ' : '') +
                                Array.prototype.slice.call(arguments, 2, 2 + n).join(', '));
                        };
                imports.Math = imports.Math || Math;
                imports.Date = imports.Date || Date;
                return baseModule;
            }
            function postInstantiate(baseModule, instance) {
                const rawExports = instance.exports;
                const memory = rawExports.memory;
                const table = rawExports.table;
                const alloc2 = rawExports['__alloc'];
                const retain = rawExports['__retain'];
                const rttiBase = rawExports['__rtti_base'] || ~0;
                function getInfo(id) {
                    const U32 = new Uint32Array(memory.buffer);
                    const count = U32[rttiBase >>> 2];
                    if ((id >>>= 0) >= count)
                        throw Error('invalid id: ' + id);
                    return U32[((rttiBase + 4) >>> 2) + id * 2];
                }
                function getBase(id) {
                    const U32 = new Uint32Array(memory.buffer);
                    const count = U32[rttiBase >>> 2];
                    if ((id >>>= 0) >= count)
                        throw Error('invalid id: ' + id);
                    return U32[((rttiBase + 4) >>> 2) + id * 2 + 1];
                }
                function getValueAlign(info) {
                    return 31 - Math.clz32((info >>> VAL_ALIGN_OFFSET) & 31);
                }
                function getKeyAlign(info) {
                    return 31 - Math.clz32((info >>> KEY_ALIGN_OFFSET) & 31);
                }
                function __allocString(str) {
                    const length2 = str.length;
                    const ptr = alloc2(length2 << 1, STRING_ID);
                    const U16 = new Uint16Array(memory.buffer);
                    for (var i = 0, p = ptr >>> 1; i < length2; ++i)
                        U16[p + i] = str.charCodeAt(i);
                    return ptr;
                }
                baseModule.__allocString = __allocString;
                function __getString(ptr) {
                    const buffer = memory.buffer;
                    const id = new Uint32Array(buffer)[(ptr + ID_OFFSET) >>> 2];
                    if (id !== STRING_ID)
                        throw Error('not a string: ' + ptr);
                    return getStringImpl(buffer, ptr);
                }
                baseModule.__getString = __getString;
                function getView(alignLog2, signed, float) {
                    const buffer = memory.buffer;
                    if (float) {
                        switch (alignLog2) {
                            case 2:
                                return new Float32Array(buffer);
                            case 3:
                                return new Float64Array(buffer);
                        }
                    }
                    else {
                        switch (alignLog2) {
                            case 0:
                                return new (signed ? Int8Array : Uint8Array)(buffer);
                            case 1:
                                return new (signed ? Int16Array : Uint16Array)(buffer);
                            case 2:
                                return new (signed ? Int32Array : Uint32Array)(buffer);
                            case 3:
                                return new (signed ? BigInt64Array : BigUint64Array)(buffer);
                        }
                    }
                    throw Error('unsupported align: ' + alignLog2);
                }
                function __allocArray(id, values) {
                    const info = getInfo(id);
                    if (!(info & (ARRAYBUFFERVIEW | ARRAY)))
                        throw Error('not an array: ' + id + ' @ ' + info);
                    const align = getValueAlign(info);
                    const length2 = values.length;
                    const buf = alloc2(length2 << align, ARRAYBUFFER_ID);
                    const arr = alloc2(info & ARRAY ? ARRAY_SIZE : ARRAYBUFFERVIEW_SIZE, id);
                    const U32 = new Uint32Array(memory.buffer);
                    U32[(arr + ARRAYBUFFERVIEW_BUFFER_OFFSET) >>> 2] = retain(buf);
                    U32[(arr + ARRAYBUFFERVIEW_DATASTART_OFFSET) >>> 2] = buf;
                    U32[(arr + ARRAYBUFFERVIEW_DATALENGTH_OFFSET) >>> 2] =
                        length2 << align;
                    if (info & ARRAY)
                        U32[(arr + ARRAY_LENGTH_OFFSET) >>> 2] = length2;
                    const view = getView(align, info & VAL_SIGNED, info & VAL_FLOAT);
                    if (info & VAL_MANAGED) {
                        for (let i = 0; i < length2; ++i)
                            view[(buf >>> align) + i] = retain(values[i]);
                    }
                    else {
                        view.set(values, buf >>> align);
                    }
                    return arr;
                }
                baseModule.__allocArray = __allocArray;
                function __getArrayView(arr) {
                    const U32 = new Uint32Array(memory.buffer);
                    const id = U32[(arr + ID_OFFSET) >>> 2];
                    const info = getInfo(id);
                    if (!(info & ARRAYBUFFERVIEW))
                        throw Error('not an array: ' + id);
                    const align = getValueAlign(info);
                    var buf = U32[(arr + ARRAYBUFFERVIEW_DATASTART_OFFSET) >>> 2];
                    const length2 = info & ARRAY
                        ? U32[(arr + ARRAY_LENGTH_OFFSET) >>> 2]
                        : U32[(buf + SIZE_OFFSET) >>> 2] >>> align;
                    return getView(align, info & VAL_SIGNED, info & VAL_FLOAT).subarray((buf >>>= align), buf + length2);
                }
                baseModule.__getArrayView = __getArrayView;
                function __getArray(arr) {
                    const input = __getArrayView(arr);
                    const len = input.length;
                    const out = new Array(len);
                    for (let i = 0; i < len; i++)
                        out[i] = input[i];
                    return out;
                }
                baseModule.__getArray = __getArray;
                function __getArrayBuffer(ptr) {
                    const buffer = memory.buffer;
                    const length2 = new Uint32Array(buffer)[(ptr + SIZE_OFFSET) >>> 2];
                    return buffer.slice(ptr, ptr + length2);
                }
                baseModule.__getArrayBuffer = __getArrayBuffer;
                function getTypedArray(Type, alignLog2, ptr) {
                    return new Type(getTypedArrayView(Type, alignLog2, ptr));
                }
                function getTypedArrayView(Type, alignLog2, ptr) {
                    const buffer = memory.buffer;
                    const U32 = new Uint32Array(buffer);
                    const bufPtr = U32[(ptr + ARRAYBUFFERVIEW_DATASTART_OFFSET) >>> 2];
                    return new Type(buffer, bufPtr, U32[(bufPtr + SIZE_OFFSET) >>> 2] >>> alignLog2);
                }
                baseModule.__getInt8Array = getTypedArray.bind(null, Int8Array, 0);
                baseModule.__getInt8ArrayView = getTypedArrayView.bind(null, Int8Array, 0);
                baseModule.__getUint8Array = getTypedArray.bind(null, Uint8Array, 0);
                baseModule.__getUint8ArrayView = getTypedArrayView.bind(null, Uint8Array, 0);
                baseModule.__getUint8ClampedArray = getTypedArray.bind(null, Uint8ClampedArray, 0);
                baseModule.__getUint8ClampedArrayView = getTypedArrayView.bind(null, Uint8ClampedArray, 0);
                baseModule.__getInt16Array = getTypedArray.bind(null, Int16Array, 1);
                baseModule.__getInt16ArrayView = getTypedArrayView.bind(null, Int16Array, 1);
                baseModule.__getUint16Array = getTypedArray.bind(null, Uint16Array, 1);
                baseModule.__getUint16ArrayView = getTypedArrayView.bind(null, Uint16Array, 1);
                baseModule.__getInt32Array = getTypedArray.bind(null, Int32Array, 2);
                baseModule.__getInt32ArrayView = getTypedArrayView.bind(null, Int32Array, 2);
                baseModule.__getUint32Array = getTypedArray.bind(null, Uint32Array, 2);
                baseModule.__getUint32ArrayView = getTypedArrayView.bind(null, Uint32Array, 2);
                if (BIGINT) {
                    baseModule.__getInt64Array = getTypedArray.bind(null, BigInt64Array, 3);
                    baseModule.__getInt64ArrayView = getTypedArrayView.bind(null, BigInt64Array, 3);
                    baseModule.__getUint64Array = getTypedArray.bind(null, BigUint64Array, 3);
                    baseModule.__getUint64ArrayView = getTypedArrayView.bind(null, BigUint64Array, 3);
                }
                baseModule.__getFloat32Array = getTypedArray.bind(null, Float32Array, 2);
                baseModule.__getFloat32ArrayView = getTypedArrayView.bind(null, Float32Array, 2);
                baseModule.__getFloat64Array = getTypedArray.bind(null, Float64Array, 3);
                baseModule.__getFloat64ArrayView = getTypedArrayView.bind(null, Float64Array, 3);
                function __instanceof(ptr, baseId) {
                    const U32 = new Uint32Array(memory.buffer);
                    var id = U32[(ptr + ID_OFFSET) >>> 2];
                    if (id <= U32[rttiBase >>> 2]) {
                        do
                            if (id == baseId)
                                return true;
                        while ((id = getBase(id)));
                    }
                    return false;
                }
                baseModule.__instanceof = __instanceof;
                baseModule.memory = baseModule.memory || memory;
                baseModule.table = baseModule.table || table;
                return demangle(rawExports, baseModule);
            }
            function isResponse(o) {
                return typeof Response !== 'undefined' && o instanceof Response;
            }
            async function instantiate(source, imports) {
                if (isResponse((source = await source)))
                    return instantiateStreaming(source, imports);
                return postInstantiate(preInstantiate(imports || (imports = {})), await WebAssembly.instantiate(source instanceof WebAssembly.Module
                    ? source
                    : await WebAssembly.compile(source), imports));
            }
            exports2.instantiate = instantiate;
            function instantiateSync(source, imports) {
                return postInstantiate(preInstantiate(imports || (imports = {})), new WebAssembly.Instance(source instanceof WebAssembly.Module
                    ? source
                    : new WebAssembly.Module(source), imports));
            }
            exports2.instantiateSync = instantiateSync;
            async function instantiateStreaming(source, imports) {
                if (!WebAssembly.instantiateStreaming) {
                    return instantiate(isResponse((source = await source)) ? source.arrayBuffer() : source, imports);
                }
                return postInstantiate(preInstantiate(imports || (imports = {})), (await WebAssembly.instantiateStreaming(source, imports)).instance);
            }
            exports2.instantiateStreaming = instantiateStreaming;
            function demangle(exports3, baseModule) {
                var module3 = baseModule ? Object.create(baseModule) : {};
                var setArgumentsLength = exports3['__argumentsLength']
                    ? function (length2) {
                        exports3['__argumentsLength'].value = length2;
                    }
                    : exports3['__setArgumentsLength'] ||
                        exports3['__setargc'] ||
                        function () { };
                for (let internalName in exports3) {
                    if (!Object.prototype.hasOwnProperty.call(exports3, internalName))
                        continue;
                    const elem = exports3[internalName];
                    let parts = internalName.split('.');
                    let curr = module3;
                    while (parts.length > 1) {
                        let part = parts.shift();
                        if (!Object.prototype.hasOwnProperty.call(curr, part))
                            curr[part] = {};
                        curr = curr[part];
                    }
                    let name4 = parts[0];
                    let hash = name4.indexOf('#');
                    if (hash >= 0) {
                        let className = name4.substring(0, hash);
                        let classElem = curr[className];
                        if (typeof classElem === 'undefined' || !classElem.prototype) {
                            let ctor = function (...args) {
                                return ctor.wrap(ctor.prototype.constructor(0, ...args));
                            };
                            ctor.prototype = {
                                valueOf: function valueOf() {
                                    return this[THIS];
                                },
                            };
                            ctor.wrap = function (thisValue) {
                                return Object.create(ctor.prototype, {
                                    [THIS]: { value: thisValue, writable: false },
                                });
                            };
                            if (classElem)
                                Object.getOwnPropertyNames(classElem).forEach((name5) => Object.defineProperty(ctor, name5, Object.getOwnPropertyDescriptor(classElem, name5)));
                            curr[className] = ctor;
                        }
                        name4 = name4.substring(hash + 1);
                        curr = curr[className].prototype;
                        if (/^(get|set):/.test(name4)) {
                            if (!Object.prototype.hasOwnProperty.call(curr, (name4 = name4.substring(4)))) {
                                let getter = exports3[internalName.replace('set:', 'get:')];
                                let setter = exports3[internalName.replace('get:', 'set:')];
                                Object.defineProperty(curr, name4, {
                                    get: function () {
                                        return getter(this[THIS]);
                                    },
                                    set: function (value) {
                                        setter(this[THIS], value);
                                    },
                                    enumerable: true,
                                });
                            }
                        }
                        else {
                            if (name4 === 'constructor') {
                                (curr[name4] = (...args) => {
                                    setArgumentsLength(args.length);
                                    return elem(...args);
                                }).original = elem;
                            }
                            else {
                                (curr[name4] = function (...args) {
                                    setArgumentsLength(args.length);
                                    return elem(this[THIS], ...args);
                                }).original = elem;
                            }
                        }
                    }
                    else {
                        if (/^(get|set):/.test(name4)) {
                            if (!Object.prototype.hasOwnProperty.call(curr, (name4 = name4.substring(4)))) {
                                Object.defineProperty(curr, name4, {
                                    get: exports3[internalName.replace('set:', 'get:')],
                                    set: exports3[internalName.replace('get:', 'set:')],
                                    enumerable: true,
                                });
                            }
                        }
                        else if (typeof elem === 'function' &&
                            elem !== setArgumentsLength) {
                            (curr[name4] = (...args) => {
                                setArgumentsLength(args.length);
                                return elem(...args);
                            }).original = elem;
                        }
                        else {
                            curr[name4] = elem;
                        }
                    }
                }
                return module3;
            }
            exports2.demangle = demangle;
        },
    });
    // node_modules/rabin-wasm/dist/rabin-wasm.js
    var require_rabin_wasm = __commonJS({
        'node_modules/rabin-wasm/dist/rabin-wasm.js'(exports2, module2) {
            var { instantiate } = require_loader();
            loadWebAssembly.supported = typeof WebAssembly !== 'undefined';
            function loadWebAssembly(imp = {}) {
                if (!loadWebAssembly.supported)
                    return null;
                var wasm = new Uint8Array([
                    0, 97, 115, 109, 1, 0, 0, 0, 1, 78, 14, 96, 2, 127, 126, 0, 96, 1,
                    127, 1, 126, 96, 2, 127, 127, 0, 96, 1, 127, 1, 127, 96, 1, 127, 0,
                    96, 2, 127, 127, 1, 127, 96, 3, 127, 127, 127, 1, 127, 96, 0, 0, 96,
                    3, 127, 127, 127, 0, 96, 0, 1, 127, 96, 4, 127, 127, 127, 127, 0, 96,
                    5, 127, 127, 127, 127, 127, 1, 127, 96, 1, 126, 1, 127, 96, 2, 126,
                    126, 1, 126, 2, 13, 1, 3, 101, 110, 118, 5, 97, 98, 111, 114, 116, 0,
                    10, 3, 54, 53, 2, 2, 8, 9, 3, 5, 2, 8, 6, 5, 3, 4, 2, 6, 9, 12, 13, 2,
                    5, 11, 3, 2, 3, 2, 3, 2, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1,
                    0, 1, 0, 1, 0, 1, 0, 6, 7, 7, 4, 4, 5, 3, 1, 0, 1, 6, 47, 9, 127, 1,
                    65, 0, 11, 127, 1, 65, 0, 11, 127, 0, 65, 3, 11, 127, 0, 65, 4, 11,
                    127, 1, 65, 0, 11, 127, 1, 65, 0, 11, 127, 1, 65, 0, 11, 127, 0, 65,
                    240, 2, 11, 127, 0, 65, 6, 11, 7, 240, 5, 41, 6, 109, 101, 109, 111,
                    114, 121, 2, 0, 7, 95, 95, 97, 108, 108, 111, 99, 0, 10, 8, 95, 95,
                    114, 101, 116, 97, 105, 110, 0, 11, 9, 95, 95, 114, 101, 108, 101, 97,
                    115, 101, 0, 12, 9, 95, 95, 99, 111, 108, 108, 101, 99, 116, 0, 51,
                    11, 95, 95, 114, 116, 116, 105, 95, 98, 97, 115, 101, 3, 7, 13, 73,
                    110, 116, 51, 50, 65, 114, 114, 97, 121, 95, 73, 68, 3, 2, 13, 85,
                    105, 110, 116, 56, 65, 114, 114, 97, 121, 95, 73, 68, 3, 3, 6, 100,
                    101, 103, 114, 101, 101, 0, 16, 3, 109, 111, 100, 0, 17, 5, 82, 97,
                    98, 105, 110, 3, 8, 16, 82, 97, 98, 105, 110, 35, 103, 101, 116, 58,
                    119, 105, 110, 100, 111, 119, 0, 21, 16, 82, 97, 98, 105, 110, 35,
                    115, 101, 116, 58, 119, 105, 110, 100, 111, 119, 0, 22, 21, 82, 97,
                    98, 105, 110, 35, 103, 101, 116, 58, 119, 105, 110, 100, 111, 119, 95,
                    115, 105, 122, 101, 0, 23, 21, 82, 97, 98, 105, 110, 35, 115, 101,
                    116, 58, 119, 105, 110, 100, 111, 119, 95, 115, 105, 122, 101, 0, 24,
                    14, 82, 97, 98, 105, 110, 35, 103, 101, 116, 58, 119, 112, 111, 115,
                    0, 25, 14, 82, 97, 98, 105, 110, 35, 115, 101, 116, 58, 119, 112, 111,
                    115, 0, 26, 15, 82, 97, 98, 105, 110, 35, 103, 101, 116, 58, 99, 111,
                    117, 110, 116, 0, 27, 15, 82, 97, 98, 105, 110, 35, 115, 101, 116, 58,
                    99, 111, 117, 110, 116, 0, 28, 13, 82, 97, 98, 105, 110, 35, 103, 101,
                    116, 58, 112, 111, 115, 0, 29, 13, 82, 97, 98, 105, 110, 35, 115, 101,
                    116, 58, 112, 111, 115, 0, 30, 15, 82, 97, 98, 105, 110, 35, 103, 101,
                    116, 58, 115, 116, 97, 114, 116, 0, 31, 15, 82, 97, 98, 105, 110, 35,
                    115, 101, 116, 58, 115, 116, 97, 114, 116, 0, 32, 16, 82, 97, 98, 105,
                    110, 35, 103, 101, 116, 58, 100, 105, 103, 101, 115, 116, 0, 33, 16,
                    82, 97, 98, 105, 110, 35, 115, 101, 116, 58, 100, 105, 103, 101, 115,
                    116, 0, 34, 21, 82, 97, 98, 105, 110, 35, 103, 101, 116, 58, 99, 104,
                    117, 110, 107, 95, 115, 116, 97, 114, 116, 0, 35, 21, 82, 97, 98, 105,
                    110, 35, 115, 101, 116, 58, 99, 104, 117, 110, 107, 95, 115, 116, 97,
                    114, 116, 0, 36, 22, 82, 97, 98, 105, 110, 35, 103, 101, 116, 58, 99,
                    104, 117, 110, 107, 95, 108, 101, 110, 103, 116, 104, 0, 37, 22, 82,
                    97, 98, 105, 110, 35, 115, 101, 116, 58, 99, 104, 117, 110, 107, 95,
                    108, 101, 110, 103, 116, 104, 0, 38, 31, 82, 97, 98, 105, 110, 35,
                    103, 101, 116, 58, 99, 104, 117, 110, 107, 95, 99, 117, 116, 95, 102,
                    105, 110, 103, 101, 114, 112, 114, 105, 110, 116, 0, 39, 31, 82, 97,
                    98, 105, 110, 35, 115, 101, 116, 58, 99, 104, 117, 110, 107, 95, 99,
                    117, 116, 95, 102, 105, 110, 103, 101, 114, 112, 114, 105, 110, 116,
                    0, 40, 20, 82, 97, 98, 105, 110, 35, 103, 101, 116, 58, 112, 111, 108,
                    121, 110, 111, 109, 105, 97, 108, 0, 41, 20, 82, 97, 98, 105, 110, 35,
                    115, 101, 116, 58, 112, 111, 108, 121, 110, 111, 109, 105, 97, 108, 0,
                    42, 17, 82, 97, 98, 105, 110, 35, 103, 101, 116, 58, 109, 105, 110,
                    115, 105, 122, 101, 0, 43, 17, 82, 97, 98, 105, 110, 35, 115, 101,
                    116, 58, 109, 105, 110, 115, 105, 122, 101, 0, 44, 17, 82, 97, 98,
                    105, 110, 35, 103, 101, 116, 58, 109, 97, 120, 115, 105, 122, 101, 0,
                    45, 17, 82, 97, 98, 105, 110, 35, 115, 101, 116, 58, 109, 97, 120,
                    115, 105, 122, 101, 0, 46, 14, 82, 97, 98, 105, 110, 35, 103, 101,
                    116, 58, 109, 97, 115, 107, 0, 47, 14, 82, 97, 98, 105, 110, 35, 115,
                    101, 116, 58, 109, 97, 115, 107, 0, 48, 17, 82, 97, 98, 105, 110, 35,
                    99, 111, 110, 115, 116, 114, 117, 99, 116, 111, 114, 0, 20, 17, 82,
                    97, 98, 105, 110, 35, 102, 105, 110, 103, 101, 114, 112, 114, 105,
                    110, 116, 0, 49, 8, 1, 50, 10, 165, 31, 53, 199, 1, 1, 4, 127, 32, 1,
                    40, 2, 0, 65, 124, 113, 34, 2, 65, 128, 2, 73, 4, 127, 32, 2, 65, 4,
                    118, 33, 4, 65, 0, 5, 32, 2, 65, 31, 32, 2, 103, 107, 34, 3, 65, 4,
                    107, 118, 65, 16, 115, 33, 4, 32, 3, 65, 7, 107, 11, 33, 3, 32, 1, 40,
                    2, 20, 33, 2, 32, 1, 40, 2, 16, 34, 5, 4, 64, 32, 5, 32, 2, 54, 2, 20,
                    11, 32, 2, 4, 64, 32, 2, 32, 5, 54, 2, 16, 11, 32, 1, 32, 0, 32, 4,
                    32, 3, 65, 4, 116, 106, 65, 2, 116, 106, 40, 2, 96, 70, 4, 64, 32, 0,
                    32, 4, 32, 3, 65, 4, 116, 106, 65, 2, 116, 106, 32, 2, 54, 2, 96, 32,
                    2, 69, 4, 64, 32, 0, 32, 3, 65, 2, 116, 106, 32, 0, 32, 3, 65, 2, 116,
                    106, 40, 2, 4, 65, 1, 32, 4, 116, 65, 127, 115, 113, 34, 1, 54, 2, 4,
                    32, 1, 69, 4, 64, 32, 0, 32, 0, 40, 2, 0, 65, 1, 32, 3, 116, 65, 127,
                    115, 113, 54, 2, 0, 11, 11, 11, 11, 226, 2, 1, 6, 127, 32, 1, 40, 2,
                    0, 33, 3, 32, 1, 65, 16, 106, 32, 1, 40, 2, 0, 65, 124, 113, 106, 34,
                    4, 40, 2, 0, 34, 5, 65, 1, 113, 4, 64, 32, 3, 65, 124, 113, 65, 16,
                    106, 32, 5, 65, 124, 113, 106, 34, 2, 65, 240, 255, 255, 255, 3, 73,
                    4, 64, 32, 0, 32, 4, 16, 1, 32, 1, 32, 2, 32, 3, 65, 3, 113, 114, 34,
                    3, 54, 2, 0, 32, 1, 65, 16, 106, 32, 1, 40, 2, 0, 65, 124, 113, 106,
                    34, 4, 40, 2, 0, 33, 5, 11, 11, 32, 3, 65, 2, 113, 4, 64, 32, 1, 65,
                    4, 107, 40, 2, 0, 34, 2, 40, 2, 0, 34, 6, 65, 124, 113, 65, 16, 106,
                    32, 3, 65, 124, 113, 106, 34, 7, 65, 240, 255, 255, 255, 3, 73, 4, 64,
                    32, 0, 32, 2, 16, 1, 32, 2, 32, 7, 32, 6, 65, 3, 113, 114, 34, 3, 54,
                    2, 0, 32, 2, 33, 1, 11, 11, 32, 4, 32, 5, 65, 2, 114, 54, 2, 0, 32, 4,
                    65, 4, 107, 32, 1, 54, 2, 0, 32, 0, 32, 3, 65, 124, 113, 34, 2, 65,
                    128, 2, 73, 4, 127, 32, 2, 65, 4, 118, 33, 4, 65, 0, 5, 32, 2, 65, 31,
                    32, 2, 103, 107, 34, 2, 65, 4, 107, 118, 65, 16, 115, 33, 4, 32, 2,
                    65, 7, 107, 11, 34, 3, 65, 4, 116, 32, 4, 106, 65, 2, 116, 106, 40, 2,
                    96, 33, 2, 32, 1, 65, 0, 54, 2, 16, 32, 1, 32, 2, 54, 2, 20, 32, 2, 4,
                    64, 32, 2, 32, 1, 54, 2, 16, 11, 32, 0, 32, 4, 32, 3, 65, 4, 116, 106,
                    65, 2, 116, 106, 32, 1, 54, 2, 96, 32, 0, 32, 0, 40, 2, 0, 65, 1, 32,
                    3, 116, 114, 54, 2, 0, 32, 0, 32, 3, 65, 2, 116, 106, 32, 0, 32, 3,
                    65, 2, 116, 106, 40, 2, 4, 65, 1, 32, 4, 116, 114, 54, 2, 4, 11, 119,
                    1, 1, 127, 32, 2, 2, 127, 32, 0, 40, 2, 160, 12, 34, 2, 4, 64, 32, 2,
                    32, 1, 65, 16, 107, 70, 4, 64, 32, 2, 40, 2, 0, 33, 3, 32, 1, 65, 16,
                    107, 33, 1, 11, 11, 32, 1, 11, 107, 34, 2, 65, 48, 73, 4, 64, 15, 11,
                    32, 1, 32, 3, 65, 2, 113, 32, 2, 65, 32, 107, 65, 1, 114, 114, 54, 2,
                    0, 32, 1, 65, 0, 54, 2, 16, 32, 1, 65, 0, 54, 2, 20, 32, 1, 32, 2,
                    106, 65, 16, 107, 34, 2, 65, 2, 54, 2, 0, 32, 0, 32, 2, 54, 2, 160,
                    12, 32, 0, 32, 1, 16, 2, 11, 155, 1, 1, 3, 127, 35, 0, 34, 0, 69, 4,
                    64, 65, 1, 63, 0, 34, 0, 74, 4, 127, 65, 1, 32, 0, 107, 64, 0, 65, 0,
                    72, 5, 65, 0, 11, 4, 64, 0, 11, 65, 176, 3, 34, 0, 65, 0, 54, 2, 0,
                    65, 208, 15, 65, 0, 54, 2, 0, 3, 64, 32, 1, 65, 23, 73, 4, 64, 32, 1,
                    65, 2, 116, 65, 176, 3, 106, 65, 0, 54, 2, 4, 65, 0, 33, 2, 3, 64, 32,
                    2, 65, 16, 73, 4, 64, 32, 1, 65, 4, 116, 32, 2, 106, 65, 2, 116, 65,
                    176, 3, 106, 65, 0, 54, 2, 96, 32, 2, 65, 1, 106, 33, 2, 12, 1, 11,
                    11, 32, 1, 65, 1, 106, 33, 1, 12, 1, 11, 11, 65, 176, 3, 65, 224, 15,
                    63, 0, 65, 16, 116, 16, 3, 65, 176, 3, 36, 0, 11, 32, 0, 11, 45, 0,
                    32, 0, 65, 240, 255, 255, 255, 3, 79, 4, 64, 65, 32, 65, 224, 0, 65,
                    201, 3, 65, 29, 16, 0, 0, 11, 32, 0, 65, 15, 106, 65, 112, 113, 34, 0,
                    65, 16, 32, 0, 65, 16, 75, 27, 11, 169, 1, 1, 1, 127, 32, 0, 32, 1,
                    65, 128, 2, 73, 4, 127, 32, 1, 65, 4, 118, 33, 1, 65, 0, 5, 32, 1, 65,
                    248, 255, 255, 255, 1, 73, 4, 64, 32, 1, 65, 1, 65, 27, 32, 1, 103,
                    107, 116, 106, 65, 1, 107, 33, 1, 11, 32, 1, 65, 31, 32, 1, 103, 107,
                    34, 2, 65, 4, 107, 118, 65, 16, 115, 33, 1, 32, 2, 65, 7, 107, 11, 34,
                    2, 65, 2, 116, 106, 40, 2, 4, 65, 127, 32, 1, 116, 113, 34, 1, 4, 127,
                    32, 0, 32, 1, 104, 32, 2, 65, 4, 116, 106, 65, 2, 116, 106, 40, 2, 96,
                    5, 32, 0, 40, 2, 0, 65, 127, 32, 2, 65, 1, 106, 116, 113, 34, 1, 4,
                    127, 32, 0, 32, 0, 32, 1, 104, 34, 0, 65, 2, 116, 106, 40, 2, 4, 104,
                    32, 0, 65, 4, 116, 106, 65, 2, 116, 106, 40, 2, 96, 5, 65, 0, 11, 11,
                    11, 111, 1, 1, 127, 63, 0, 34, 2, 32, 1, 65, 248, 255, 255, 255, 1,
                    73, 4, 127, 32, 1, 65, 1, 65, 27, 32, 1, 103, 107, 116, 65, 1, 107,
                    106, 5, 32, 1, 11, 65, 16, 32, 0, 40, 2, 160, 12, 32, 2, 65, 16, 116,
                    65, 16, 107, 71, 116, 106, 65, 255, 255, 3, 106, 65, 128, 128, 124,
                    113, 65, 16, 118, 34, 1, 32, 2, 32, 1, 74, 27, 64, 0, 65, 0, 72, 4,
                    64, 32, 1, 64, 0, 65, 0, 72, 4, 64, 0, 11, 11, 32, 0, 32, 2, 65, 16,
                    116, 63, 0, 65, 16, 116, 16, 3, 11, 113, 1, 2, 127, 32, 1, 40, 2, 0,
                    34, 3, 65, 124, 113, 32, 2, 107, 34, 4, 65, 32, 79, 4, 64, 32, 1, 32,
                    2, 32, 3, 65, 2, 113, 114, 54, 2, 0, 32, 2, 32, 1, 65, 16, 106, 106,
                    34, 1, 32, 4, 65, 16, 107, 65, 1, 114, 54, 2, 0, 32, 0, 32, 1, 16, 2,
                    5, 32, 1, 32, 3, 65, 126, 113, 54, 2, 0, 32, 1, 65, 16, 106, 32, 1,
                    40, 2, 0, 65, 124, 113, 106, 32, 1, 65, 16, 106, 32, 1, 40, 2, 0, 65,
                    124, 113, 106, 40, 2, 0, 65, 125, 113, 54, 2, 0, 11, 11, 91, 1, 2,
                    127, 32, 0, 32, 1, 16, 5, 34, 4, 16, 6, 34, 3, 69, 4, 64, 65, 1, 36,
                    1, 65, 0, 36, 1, 32, 0, 32, 4, 16, 6, 34, 3, 69, 4, 64, 32, 0, 32, 4,
                    16, 7, 32, 0, 32, 4, 16, 6, 33, 3, 11, 11, 32, 3, 65, 0, 54, 2, 4, 32,
                    3, 32, 2, 54, 2, 8, 32, 3, 32, 1, 54, 2, 12, 32, 0, 32, 3, 16, 1, 32,
                    0, 32, 3, 32, 4, 16, 8, 32, 3, 11, 13, 0, 16, 4, 32, 0, 32, 1, 16, 9,
                    65, 16, 106, 11, 33, 1, 1, 127, 32, 0, 65, 172, 3, 75, 4, 64, 32, 0,
                    65, 16, 107, 34, 1, 32, 1, 40, 2, 4, 65, 1, 106, 54, 2, 4, 11, 32, 0,
                    11, 18, 0, 32, 0, 65, 172, 3, 75, 4, 64, 32, 0, 65, 16, 107, 16, 52,
                    11, 11, 140, 3, 1, 1, 127, 2, 64, 32, 1, 69, 13, 0, 32, 0, 65, 0, 58,
                    0, 0, 32, 0, 32, 1, 106, 65, 1, 107, 65, 0, 58, 0, 0, 32, 1, 65, 2,
                    77, 13, 0, 32, 0, 65, 1, 106, 65, 0, 58, 0, 0, 32, 0, 65, 2, 106, 65,
                    0, 58, 0, 0, 32, 0, 32, 1, 106, 34, 2, 65, 2, 107, 65, 0, 58, 0, 0,
                    32, 2, 65, 3, 107, 65, 0, 58, 0, 0, 32, 1, 65, 6, 77, 13, 0, 32, 0,
                    65, 3, 106, 65, 0, 58, 0, 0, 32, 0, 32, 1, 106, 65, 4, 107, 65, 0, 58,
                    0, 0, 32, 1, 65, 8, 77, 13, 0, 32, 1, 65, 0, 32, 0, 107, 65, 3, 113,
                    34, 1, 107, 33, 2, 32, 0, 32, 1, 106, 34, 0, 65, 0, 54, 2, 0, 32, 0,
                    32, 2, 65, 124, 113, 34, 1, 106, 65, 4, 107, 65, 0, 54, 2, 0, 32, 1,
                    65, 8, 77, 13, 0, 32, 0, 65, 4, 106, 65, 0, 54, 2, 0, 32, 0, 65, 8,
                    106, 65, 0, 54, 2, 0, 32, 0, 32, 1, 106, 34, 2, 65, 12, 107, 65, 0,
                    54, 2, 0, 32, 2, 65, 8, 107, 65, 0, 54, 2, 0, 32, 1, 65, 24, 77, 13,
                    0, 32, 0, 65, 12, 106, 65, 0, 54, 2, 0, 32, 0, 65, 16, 106, 65, 0, 54,
                    2, 0, 32, 0, 65, 20, 106, 65, 0, 54, 2, 0, 32, 0, 65, 24, 106, 65, 0,
                    54, 2, 0, 32, 0, 32, 1, 106, 34, 2, 65, 28, 107, 65, 0, 54, 2, 0, 32,
                    2, 65, 24, 107, 65, 0, 54, 2, 0, 32, 2, 65, 20, 107, 65, 0, 54, 2, 0,
                    32, 2, 65, 16, 107, 65, 0, 54, 2, 0, 32, 0, 32, 0, 65, 4, 113, 65, 24,
                    106, 34, 2, 106, 33, 0, 32, 1, 32, 2, 107, 33, 1, 3, 64, 32, 1, 65,
                    32, 79, 4, 64, 32, 0, 66, 0, 55, 3, 0, 32, 0, 65, 8, 106, 66, 0, 55,
                    3, 0, 32, 0, 65, 16, 106, 66, 0, 55, 3, 0, 32, 0, 65, 24, 106, 66, 0,
                    55, 3, 0, 32, 1, 65, 32, 107, 33, 1, 32, 0, 65, 32, 106, 33, 0, 12, 1,
                    11, 11, 11, 11, 178, 1, 1, 3, 127, 32, 1, 65, 240, 255, 255, 255, 3,
                    32, 2, 118, 75, 4, 64, 65, 144, 1, 65, 192, 1, 65, 23, 65, 56, 16, 0,
                    0, 11, 32, 1, 32, 2, 116, 34, 3, 65, 0, 16, 10, 34, 2, 32, 3, 16, 13,
                    32, 0, 69, 4, 64, 65, 12, 65, 2, 16, 10, 34, 0, 65, 172, 3, 75, 4, 64,
                    32, 0, 65, 16, 107, 34, 1, 32, 1, 40, 2, 4, 65, 1, 106, 54, 2, 4, 11,
                    11, 32, 0, 65, 0, 54, 2, 0, 32, 0, 65, 0, 54, 2, 4, 32, 0, 65, 0, 54,
                    2, 8, 32, 2, 34, 1, 32, 0, 40, 2, 0, 34, 4, 71, 4, 64, 32, 1, 65, 172,
                    3, 75, 4, 64, 32, 1, 65, 16, 107, 34, 5, 32, 5, 40, 2, 4, 65, 1, 106,
                    54, 2, 4, 11, 32, 4, 16, 12, 11, 32, 0, 32, 1, 54, 2, 0, 32, 0, 32, 2,
                    54, 2, 4, 32, 0, 32, 3, 54, 2, 8, 32, 0, 11, 46, 1, 2, 127, 65, 12,
                    65, 5, 16, 10, 34, 0, 65, 172, 3, 75, 4, 64, 32, 0, 65, 16, 107, 34,
                    1, 32, 1, 40, 2, 4, 65, 1, 106, 54, 2, 4, 11, 32, 0, 65, 128, 2, 65,
                    3, 16, 14, 11, 9, 0, 65, 63, 32, 0, 121, 167, 107, 11, 49, 1, 2, 127,
                    65, 63, 32, 1, 121, 167, 107, 33, 2, 3, 64, 65, 63, 32, 0, 121, 167,
                    107, 32, 2, 107, 34, 3, 65, 0, 78, 4, 64, 32, 0, 32, 1, 32, 3, 172,
                    134, 133, 33, 0, 12, 1, 11, 11, 32, 0, 11, 40, 0, 32, 1, 32, 0, 40, 2,
                    8, 79, 4, 64, 65, 128, 2, 65, 192, 2, 65, 163, 1, 65, 44, 16, 0, 0,
                    11, 32, 1, 32, 0, 40, 2, 4, 106, 65, 0, 58, 0, 0, 11, 38, 0, 32, 1,
                    32, 0, 40, 2, 8, 79, 4, 64, 65, 128, 2, 65, 192, 2, 65, 152, 1, 65,
                    44, 16, 0, 0, 11, 32, 1, 32, 0, 40, 2, 4, 106, 45, 0, 0, 11, 254, 5,
                    2, 1, 127, 4, 126, 32, 0, 69, 4, 64, 65, 232, 0, 65, 6, 16, 10, 34, 0,
                    65, 172, 3, 75, 4, 64, 32, 0, 65, 16, 107, 34, 5, 32, 5, 40, 2, 4, 65,
                    1, 106, 54, 2, 4, 11, 11, 32, 0, 65, 0, 54, 2, 0, 32, 0, 65, 0, 54, 2,
                    4, 32, 0, 65, 0, 54, 2, 8, 32, 0, 66, 0, 55, 3, 16, 32, 0, 66, 0, 55,
                    3, 24, 32, 0, 66, 0, 55, 3, 32, 32, 0, 66, 0, 55, 3, 40, 32, 0, 66, 0,
                    55, 3, 48, 32, 0, 66, 0, 55, 3, 56, 32, 0, 66, 0, 55, 3, 64, 32, 0,
                    66, 0, 55, 3, 72, 32, 0, 66, 0, 55, 3, 80, 32, 0, 66, 0, 55, 3, 88,
                    32, 0, 66, 0, 55, 3, 96, 32, 0, 32, 2, 173, 55, 3, 80, 32, 0, 32, 3,
                    173, 55, 3, 88, 65, 12, 65, 4, 16, 10, 34, 2, 65, 172, 3, 75, 4, 64,
                    32, 2, 65, 16, 107, 34, 3, 32, 3, 40, 2, 4, 65, 1, 106, 54, 2, 4, 11,
                    32, 2, 32, 4, 65, 0, 16, 14, 33, 2, 32, 0, 40, 2, 0, 16, 12, 32, 0,
                    32, 2, 54, 2, 0, 32, 0, 32, 4, 54, 2, 4, 32, 0, 66, 1, 32, 1, 173,
                    134, 66, 1, 125, 55, 3, 96, 32, 0, 66, 243, 130, 183, 218, 216, 230,
                    232, 30, 55, 3, 72, 35, 4, 69, 4, 64, 65, 0, 33, 2, 3, 64, 32, 2, 65,
                    128, 2, 72, 4, 64, 32, 2, 65, 255, 1, 113, 173, 33, 6, 32, 0, 41, 3,
                    72, 34, 7, 33, 8, 65, 63, 32, 7, 121, 167, 107, 33, 1, 3, 64, 65, 63,
                    32, 6, 121, 167, 107, 32, 1, 107, 34, 3, 65, 0, 78, 4, 64, 32, 6, 32,
                    8, 32, 3, 172, 134, 133, 33, 6, 12, 1, 11, 11, 65, 0, 33, 4, 3, 64,
                    32, 4, 32, 0, 40, 2, 4, 65, 1, 107, 72, 4, 64, 32, 6, 66, 8, 134, 33,
                    6, 32, 0, 41, 3, 72, 34, 7, 33, 8, 65, 63, 32, 7, 121, 167, 107, 33,
                    1, 3, 64, 65, 63, 32, 6, 121, 167, 107, 32, 1, 107, 34, 3, 65, 0, 78,
                    4, 64, 32, 6, 32, 8, 32, 3, 172, 134, 133, 33, 6, 12, 1, 11, 11, 32,
                    4, 65, 1, 106, 33, 4, 12, 1, 11, 11, 35, 6, 40, 2, 4, 32, 2, 65, 3,
                    116, 106, 32, 6, 55, 3, 0, 32, 2, 65, 1, 106, 33, 2, 12, 1, 11, 11,
                    65, 63, 32, 0, 41, 3, 72, 121, 167, 107, 172, 33, 7, 65, 0, 33, 2, 3,
                    64, 32, 2, 65, 128, 2, 72, 4, 64, 35, 5, 33, 1, 32, 2, 172, 32, 7,
                    134, 34, 8, 33, 6, 65, 63, 32, 0, 41, 3, 72, 34, 9, 121, 167, 107, 33,
                    3, 3, 64, 65, 63, 32, 6, 121, 167, 107, 32, 3, 107, 34, 4, 65, 0, 78,
                    4, 64, 32, 6, 32, 9, 32, 4, 172, 134, 133, 33, 6, 12, 1, 11, 11, 32,
                    1, 40, 2, 4, 32, 2, 65, 3, 116, 106, 32, 6, 32, 8, 132, 55, 3, 0, 32,
                    2, 65, 1, 106, 33, 2, 12, 1, 11, 11, 65, 1, 36, 4, 11, 32, 0, 66, 0,
                    55, 3, 24, 32, 0, 66, 0, 55, 3, 32, 65, 0, 33, 2, 3, 64, 32, 2, 32, 0,
                    40, 2, 4, 72, 4, 64, 32, 0, 40, 2, 0, 32, 2, 16, 18, 32, 2, 65, 1,
                    106, 33, 2, 12, 1, 11, 11, 32, 0, 66, 0, 55, 3, 40, 32, 0, 65, 0, 54,
                    2, 8, 32, 0, 66, 0, 55, 3, 16, 32, 0, 66, 0, 55, 3, 40, 32, 0, 40, 2,
                    0, 32, 0, 40, 2, 8, 16, 19, 33, 1, 32, 0, 40, 2, 8, 32, 0, 40, 2, 0,
                    40, 2, 4, 106, 65, 1, 58, 0, 0, 32, 0, 32, 0, 41, 3, 40, 35, 6, 40, 2,
                    4, 32, 1, 65, 3, 116, 106, 41, 3, 0, 133, 55, 3, 40, 32, 0, 32, 0, 40,
                    2, 8, 65, 1, 106, 32, 0, 40, 2, 4, 111, 54, 2, 8, 32, 0, 35, 5, 40, 2,
                    4, 32, 0, 41, 3, 40, 34, 6, 66, 45, 136, 167, 65, 3, 116, 106, 41, 3,
                    0, 32, 6, 66, 8, 134, 66, 1, 132, 133, 55, 3, 40, 32, 0, 11, 38, 1, 1,
                    127, 32, 0, 40, 2, 0, 34, 0, 65, 172, 3, 75, 4, 64, 32, 0, 65, 16,
                    107, 34, 1, 32, 1, 40, 2, 4, 65, 1, 106, 54, 2, 4, 11, 32, 0, 11, 55,
                    1, 2, 127, 32, 1, 32, 0, 40, 2, 0, 34, 2, 71, 4, 64, 32, 1, 65, 172,
                    3, 75, 4, 64, 32, 1, 65, 16, 107, 34, 3, 32, 3, 40, 2, 4, 65, 1, 106,
                    54, 2, 4, 11, 32, 2, 16, 12, 11, 32, 0, 32, 1, 54, 2, 0, 11, 7, 0, 32,
                    0, 40, 2, 4, 11, 9, 0, 32, 0, 32, 1, 54, 2, 4, 11, 7, 0, 32, 0, 40, 2,
                    8, 11, 9, 0, 32, 0, 32, 1, 54, 2, 8, 11, 7, 0, 32, 0, 41, 3, 16, 11,
                    9, 0, 32, 0, 32, 1, 55, 3, 16, 11, 7, 0, 32, 0, 41, 3, 24, 11, 9, 0,
                    32, 0, 32, 1, 55, 3, 24, 11, 7, 0, 32, 0, 41, 3, 32, 11, 9, 0, 32, 0,
                    32, 1, 55, 3, 32, 11, 7, 0, 32, 0, 41, 3, 40, 11, 9, 0, 32, 0, 32, 1,
                    55, 3, 40, 11, 7, 0, 32, 0, 41, 3, 48, 11, 9, 0, 32, 0, 32, 1, 55, 3,
                    48, 11, 7, 0, 32, 0, 41, 3, 56, 11, 9, 0, 32, 0, 32, 1, 55, 3, 56, 11,
                    7, 0, 32, 0, 41, 3, 64, 11, 9, 0, 32, 0, 32, 1, 55, 3, 64, 11, 7, 0,
                    32, 0, 41, 3, 72, 11, 9, 0, 32, 0, 32, 1, 55, 3, 72, 11, 7, 0, 32, 0,
                    41, 3, 80, 11, 9, 0, 32, 0, 32, 1, 55, 3, 80, 11, 7, 0, 32, 0, 41, 3,
                    88, 11, 9, 0, 32, 0, 32, 1, 55, 3, 88, 11, 7, 0, 32, 0, 41, 3, 96, 11,
                    9, 0, 32, 0, 32, 1, 55, 3, 96, 11, 172, 4, 2, 5, 127, 1, 126, 32, 2,
                    65, 172, 3, 75, 4, 64, 32, 2, 65, 16, 107, 34, 4, 32, 4, 40, 2, 4, 65,
                    1, 106, 54, 2, 4, 11, 32, 2, 33, 4, 65, 0, 33, 2, 32, 1, 40, 2, 8, 33,
                    5, 32, 1, 40, 2, 4, 33, 6, 3, 64, 2, 127, 65, 0, 33, 3, 3, 64, 32, 3,
                    32, 5, 72, 4, 64, 32, 3, 32, 6, 106, 45, 0, 0, 33, 1, 32, 0, 40, 2, 0,
                    32, 0, 40, 2, 8, 16, 19, 33, 7, 32, 0, 40, 2, 8, 32, 0, 40, 2, 0, 40,
                    2, 4, 106, 32, 1, 58, 0, 0, 32, 0, 32, 0, 41, 3, 40, 35, 6, 40, 2, 4,
                    32, 7, 65, 3, 116, 106, 41, 3, 0, 133, 55, 3, 40, 32, 0, 32, 0, 40, 2,
                    8, 65, 1, 106, 32, 0, 40, 2, 4, 111, 54, 2, 8, 32, 0, 35, 5, 40, 2, 4,
                    32, 0, 41, 3, 40, 34, 8, 66, 45, 136, 167, 65, 3, 116, 106, 41, 3, 0,
                    32, 1, 173, 32, 8, 66, 8, 134, 132, 133, 55, 3, 40, 32, 0, 32, 0, 41,
                    3, 16, 66, 1, 124, 55, 3, 16, 32, 0, 32, 0, 41, 3, 24, 66, 1, 124, 55,
                    3, 24, 32, 0, 41, 3, 16, 32, 0, 41, 3, 80, 90, 4, 127, 32, 0, 41, 3,
                    40, 32, 0, 41, 3, 96, 131, 80, 5, 65, 0, 11, 4, 127, 65, 1, 5, 32, 0,
                    41, 3, 16, 32, 0, 41, 3, 88, 90, 11, 4, 64, 32, 0, 32, 0, 41, 3, 32,
                    55, 3, 48, 32, 0, 32, 0, 41, 3, 16, 55, 3, 56, 32, 0, 32, 0, 41, 3,
                    40, 55, 3, 64, 65, 0, 33, 1, 3, 64, 32, 1, 32, 0, 40, 2, 4, 72, 4, 64,
                    32, 0, 40, 2, 0, 32, 1, 16, 18, 32, 1, 65, 1, 106, 33, 1, 12, 1, 11,
                    11, 32, 0, 66, 0, 55, 3, 40, 32, 0, 65, 0, 54, 2, 8, 32, 0, 66, 0, 55,
                    3, 16, 32, 0, 66, 0, 55, 3, 40, 32, 0, 40, 2, 0, 32, 0, 40, 2, 8, 16,
                    19, 33, 1, 32, 0, 40, 2, 8, 32, 0, 40, 2, 0, 40, 2, 4, 106, 65, 1, 58,
                    0, 0, 32, 0, 32, 0, 41, 3, 40, 35, 6, 40, 2, 4, 32, 1, 65, 3, 116,
                    106, 41, 3, 0, 133, 55, 3, 40, 32, 0, 32, 0, 40, 2, 8, 65, 1, 106, 32,
                    0, 40, 2, 4, 111, 54, 2, 8, 32, 0, 35, 5, 40, 2, 4, 32, 0, 41, 3, 40,
                    34, 8, 66, 45, 136, 167, 65, 3, 116, 106, 41, 3, 0, 32, 8, 66, 8, 134,
                    66, 1, 132, 133, 55, 3, 40, 32, 3, 65, 1, 106, 12, 3, 11, 32, 3, 65,
                    1, 106, 33, 3, 12, 1, 11, 11, 65, 127, 11, 34, 1, 65, 0, 78, 4, 64,
                    32, 5, 32, 1, 107, 33, 5, 32, 1, 32, 6, 106, 33, 6, 32, 2, 34, 1, 65,
                    1, 106, 33, 2, 32, 4, 40, 2, 4, 32, 1, 65, 2, 116, 106, 32, 0, 41, 3,
                    56, 62, 2, 0, 12, 1, 11, 11, 32, 4, 11, 10, 0, 16, 15, 36, 5, 16, 15,
                    36, 6, 11, 3, 0, 1, 11, 73, 1, 2, 127, 32, 0, 40, 2, 4, 34, 1, 65,
                    255, 255, 255, 255, 0, 113, 34, 2, 65, 1, 70, 4, 64, 32, 0, 65, 16,
                    106, 16, 53, 32, 0, 32, 0, 40, 2, 0, 65, 1, 114, 54, 2, 0, 35, 0, 32,
                    0, 16, 2, 5, 32, 0, 32, 2, 65, 1, 107, 32, 1, 65, 128, 128, 128, 128,
                    127, 113, 114, 54, 2, 4, 11, 11, 58, 0, 2, 64, 2, 64, 2, 64, 32, 0,
                    65, 8, 107, 40, 2, 0, 14, 7, 0, 0, 1, 1, 1, 1, 1, 2, 11, 15, 11, 32,
                    0, 40, 2, 0, 34, 0, 4, 64, 32, 0, 65, 172, 3, 79, 4, 64, 32, 0, 65,
                    16, 107, 16, 52, 11, 11, 15, 11, 0, 11, 11, 137, 3, 7, 0, 65, 16, 11,
                    55, 40, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 40, 0, 0, 0, 97, 0, 108, 0,
                    108, 0, 111, 0, 99, 0, 97, 0, 116, 0, 105, 0, 111, 0, 110, 0, 32, 0,
                    116, 0, 111, 0, 111, 0, 32, 0, 108, 0, 97, 0, 114, 0, 103, 0, 101, 0,
                    65, 208, 0, 11, 45, 30, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 30, 0, 0, 0,
                    126, 0, 108, 0, 105, 0, 98, 0, 47, 0, 114, 0, 116, 0, 47, 0, 116, 0,
                    108, 0, 115, 0, 102, 0, 46, 0, 116, 0, 115, 0, 65, 128, 1, 11, 43, 28,
                    0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 28, 0, 0, 0, 73, 0, 110, 0, 118, 0,
                    97, 0, 108, 0, 105, 0, 100, 0, 32, 0, 108, 0, 101, 0, 110, 0, 103, 0,
                    116, 0, 104, 0, 65, 176, 1, 11, 53, 38, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
                    0, 38, 0, 0, 0, 126, 0, 108, 0, 105, 0, 98, 0, 47, 0, 97, 0, 114, 0,
                    114, 0, 97, 0, 121, 0, 98, 0, 117, 0, 102, 0, 102, 0, 101, 0, 114, 0,
                    46, 0, 116, 0, 115, 0, 65, 240, 1, 11, 51, 36, 0, 0, 0, 1, 0, 0, 0, 1,
                    0, 0, 0, 36, 0, 0, 0, 73, 0, 110, 0, 100, 0, 101, 0, 120, 0, 32, 0,
                    111, 0, 117, 0, 116, 0, 32, 0, 111, 0, 102, 0, 32, 0, 114, 0, 97, 0,
                    110, 0, 103, 0, 101, 0, 65, 176, 2, 11, 51, 36, 0, 0, 0, 1, 0, 0, 0,
                    1, 0, 0, 0, 36, 0, 0, 0, 126, 0, 108, 0, 105, 0, 98, 0, 47, 0, 116, 0,
                    121, 0, 112, 0, 101, 0, 100, 0, 97, 0, 114, 0, 114, 0, 97, 0, 121, 0,
                    46, 0, 116, 0, 115, 0, 65, 240, 2, 11, 53, 7, 0, 0, 0, 16, 0, 0, 0, 0,
                    0, 0, 0, 16, 0, 0, 0, 0, 0, 0, 0, 16, 0, 0, 0, 0, 0, 0, 0, 145, 4, 0,
                    0, 2, 0, 0, 0, 49, 0, 0, 0, 2, 0, 0, 0, 17, 1, 0, 0, 2, 0, 0, 0, 16,
                    0, 34, 16, 115, 111, 117, 114, 99, 101, 77, 97, 112, 112, 105, 110,
                    103, 85, 82, 76, 16, 46, 47, 114, 97, 98, 105, 110, 46, 119, 97, 115,
                    109, 46, 109, 97, 112,
                ]);
                return instantiate(new Response(new Blob([wasm], { type: 'application/wasm' })), imp);
            }
            module2.exports = loadWebAssembly;
        },
    });
    // node_modules/rabin-wasm/src/index.js
    var require_src = __commonJS({
        'node_modules/rabin-wasm/src/index.js'(exports2, module2) {
            var Rabin = require_rabin();
            var getRabin = require_rabin_wasm();
            var create3 = async (avg, min, max, windowSize, polynomial) => {
                const compiled = await getRabin();
                return new Rabin(compiled, avg, min, max, windowSize, polynomial);
            };
            module2.exports = {
                Rabin,
                create: create3,
            };
        },
    });
    // node_modules/sparse-array/index.js
    var require_sparse_array = __commonJS({
        'node_modules/sparse-array/index.js'(exports2, module2) {
            'use strict';
            var BITS_PER_BYTE = 7;
            module2.exports = class SparseArray {
                constructor() {
